====== General table of content ======
[[General TODO]]
[[Mails]]
[[To learn and read]]
[[JAK2]]
[[General_Ideas]]
[[PaperReading]]
[[KULeuven information]]
[[TextPrediction]]
[[Paper Writing]]
[[Web development]]
[[Various]]
[[Key Discovery]]

* [ ] Pattern Mining Problems
  * [ ] mining patterns in the presence of a continuous target value
  * [ ] inhibitor mining
  * [ ] combining QSAR + toxicophores
  * [ ] problems related to the super-imposition of patterns
    * [ ] introducing a notion of distance between 2 fragments of a pattern
    * [ ] need for extracting emerging patterns with a variable growth-rate (or other measures) given the coverage --> see approaches with branch-and-bound search (Direct Discriminative Pattern Mining for Effective Classification)
     [ ] mine the correlation between the toxicity (or other target attribute) and the number of occurrences of a fragment

* [ ] RPMP improving
  * [ ] could be interesting to mine closed itemsets by taking into account an existing order over the items
    * [ ] e.g., we do not want to output itemsets of the form {i1,i2...,in} where there exist j and k s.t. j < k
    * [ ] with growth rate computation, those patterns are redundant with their minimal version
    * [ ] the main principle would be: 

== Daily TODO ==

2013-02-22 Fri 09:52:47 CET
* [.] Mutagenicity
  * [ ] Run some tests on each source of the benchmark data
    * [ ] build a different dataset for each source
    * [ ] run some experiments on them
    * [ ] compare the results
  * [X] Study the correlation of the leaves' p-value with their real quality
    * [X] generate a plot with individual error rate of leaves according to their p-value
    * [X] add a linear regression fitting model
    * [X] compute a correlation coefficient between the p-value and the generalization power of leaves
* [ ] Geometrical pattern mining
  * [ ] how to mine closed patterns?
    * [ ] when reaching a leaf?
    * [ ] at each best test selection?
      * [ ] a best test is selected and we want to extend it with every refinements that do not modify its extension
  * [ ] progress on the generation of size-2 patterns
    * [ ] generate the distances
* [X] Experiments
    * [X] work on why some experiments do not compile
    * [X] re-run the tests
* [ ] General
  * [ ] write dr

2013-02-21 Thu 09:14:07 CET
* [X] Mutagenicity
  * [X] write and send a document presenting the results obtained with Mutagenicity data
    * [X] presentation of the p-value computation
    * [X] presentation of the performed studies
    * [X] presentation of the prediction strategies
  * [X] progress on the generation of size-2 patterns
* [ ] Geometrical pattern mining
  * [ ] how to mine closed patterns?

2013-02-20 Wed 10:36:51 CET
* [X] Mutagenicity
  * [X] send to Jérémie the papers describing the benchmark data sources
  * [X] write and send a Skype meeting report
  * [X] write a document presenting the results obtained with Mutagenicity data
* [X] Geometrical pattern mining
  * [X] progress on the generation of size-2 patterns
    * [X] generate the distances
* [X] Experiments
    * [X] run new tests with a larger minimum leaf size and a larger test sampling rate
* [X] General
  * [X] write dr

2013-02-19 Tue 09:07:51 CET
* [X] Project Meeting (2h15)
* [X] Skype Meeting (Jan Ramon, Bertrand Cuissart, Bruno Crémilleux) (1h)
* [X] Project Meeting presentation
  * [X] quickly read the Ames test description
  * [X] print the paper related to the benchmark dataset
  * [X] print the project description
  * [X] copy files locally
  * [X] to do to finish the presentation
    * [X] modify title slide
      * [X] find a title
    * [X] for decision tree examples, replace ic50 by Mutagenicity
    * [X] for RPMP, add some remarks about what has been done
    * [X] add colors to the text
    * [X] change 'forêt décisionnelle' for 'forêt aléatoire'
    * [X] add a sub-section about pattern explanations
  * [X] copy the presentation on USB key

2013-02-18 Mon 08:20:08 CET
* [X] Send an email to Alban Lepailleur about his presentation in the Chemoinformatics work group
* [X] RPMP mining
  * [X] use LCM via Guillaume Poezevara's work
    * [X] use an infinite growth rate
  * [X] give examples 
* [X] Project Meeting presentation
  * [X] write down the plan of the presentation
    * [X] Geometrical patterns
      * [X] explanations
        * [X] geometrical + chemical features
        * [X] random forests
      * [X] first results on Mutagenicity data
        * [X] quantitative results
        * [X] in-depth analysis and examples of plots
        * [X] examples of patterns extracted
    * [X] RPMP
      * [X] explanations
      * [X] first results on Mutagenicity data
      * [X] pattern examples (or not)
    * [X] Future work
      * [X] combination of the two pattern languages
      * [X] improving with molecular descriptors (some are provided with the Benchmark dataset)
  * [X] to do to finish the presentation
    * [X] modify title slide
      * [X] find a title
      * [X] add the logos for CERMN and ADNTox
    * [X] modify the real example to show the odds
    * [X] add source for the RPMP slides
    * [X] modify the generated plots
      * [X] add the axis labels
      * [X] generate a new plots on the best experiment
      * [X] generate a new plot with histogram with p-value between 0 and 0.1 
      * [X] integrate the new plots into the presentation
      * [X] add a density plot for good leaves
      * [X] add the error rate plot
    * [X] put the 'data description' slide right before the real example
    * [X] remove the additional 'conclusion' section
* [X] General
  * [X] send PhD thesis latex template to Hai and Hugo
  * [X] answer parents' email

2013-02-17 Sun 11:02:21 CET
* [X] RPMP mining
  * [X] find datasets of frequent subgraphs as provided by Guillaume Poezevara's code
  * [X] manually try to run the DDPMine algorithm on these data
    * [X] convert the data into DDPMine format
    * [X] run the python code --> does not work properly
* [X] Mutagenesis data analysis
  * [X] compute the various plots and measures by aggregating all the experiments
    * [X] read the experiments having the same settings but being part of a different cross-validation fold
      * [X] compute the mean and standard deviation for each measure
    * [X] read the experiments corresponding to the same settings but different random forest size
    * [X] write the global results in a csv file
* [X] General
  * [X] write dr

2013-02-16 Sat 16:39:24 CET
* [X] Mutagenesis data analysis
  * [X] compute predictions over the random forests
    * [X] average value
    * [X] average with a p-value maximum treshold
    * [X] average with a p-value maximum treshold on the leaves covering each molecule
  * [X] compute AUROC measure over the random forests
  * [X] test different computations of the prediction for a test molecule

2013-02-15 Fri 10:06:51 CET
* [X] Mutagenesis data analysis
  * [X] run new experiments with a test sampling rate of 10% rather than 5%
  * [X] run new experiments with a minimum leaf size of 20 rather than 10
  * [X] fix problem with storage of learning and test occurrences
  * [X] generate some plots:
    * [X] histogram of the number of leaves compared w.r.t. the p-value
    * [X] plot the coverage rate against the p-value maximum treshold
    * [X] plot the error rate against the p-value maximum treshold
    * [X] plot the multi-coverage of the molecules according to a p-value maximum threshold
    * [X] operations to perform for computing the plots
      * [X] calculate, for each molecule, how many times it is covered by a good leaf
  * [X] test on a chosen experiment (id 251)
  * [X] fix issues in computing the various measures
    * [X] the molecules seem to be counted two times instead of once (e.g., the maximum value in the learning dataset is around 12000 instead of around 6000)
    * [X] the multi-coverage seems wrong
      * [X] with a pvalue of 1, the count of should be around 6500

2013-02-14 Thu 10:22:40 CET
* [X] Jak2 code
  * [X] should fix a problem when looking for the molecules falling into a node
* [X] Answer Jan Ramon's email
* [X] Mutagenesis data analysis
  * [X] run the experiments again with finer information about the embeddings
* [X] General
  * [X] write dr

2013-02-13 Wed 10:04:49 CET
* [X] Results analysis
  * [X] put new decision trees online for Ronan Bureau
    * [X] for each experiment, create a file containing the decision trees and a file containing the occurrences
  * [X] send an email to Ronan and write some explanations about how to read the results (1h)
* [X] Jak2 code
  * [X] in the decision forests, fix the fact that the predicted values are wrong because of the weighting of the individual values
* [X] Mutagenesis data analysis
  * [X] learn some random forests with parameters:
    * [X] minimum leaf size: 10
    * [X] random forest size: 25, 50, 75, 100
  * [X] find a way to compute p-values in ruby scripts
  * [X] generate some plots:
    * [X] operations to perform for computing the plots
      * [X] read a random forest
      * [X] for each node, calculate a p-value
        * [X] need the number of molecules in class 0 and in class 1 (in the learning set)
        * [X] need the number of molecules in a leaf with class 0 and class 1
      * [X] filter the read nodes in order to have only the leaves
      * [X] print the leaves, sorted by the descending p-value
      * [X] for each leaf, consider the number of classification errors
      * [X] compute the error rate for the global decision tree
* [X] General
  * [X] copy public key to mobile phone
  * [X] change password for KULeuven
  * [X] write dr

2013-02-12 Tue 09:25:06 CET
* [X] Skype Meeting (Jan Ramon, Bertrand Cuissart, Bruno Crémilleux) (1h)
* [X] Meeting with Bertrand Cuissart (1h30)
  * [X] defining some quality measures for trees and patterns
* [X] Chemoinformatics meeting (1h30)
* [o] Ronan
  * [X] save the extension of leaves in training and test sets
  * [X] run specific experiments for obtaining new trees on jak2 and Mutagenicity datasets
  * [ ] put new trees on dropbox
  * [ ] Jak2 data
    * [ ] use the decoys in a 3-class problem
      * [ ] run as a regression problem and use a script to compute interesting statistics
  * [ ] generate html files for summarizing a decision tree
    * [ ] main page is the tree description
    * [ ] each leaf node points to a file describing the pattern and its extensions (in training and test sets)
      * [ ] how to fully describe a pattern?
        * [ ] from the node
          * [ ] find the ancestors
            * [ ] for each "no" node...
            * [ ] for each "yes" node...
* [ ] General
  * [ ] write dr

2013-02-11 Mon 09:24:17 CET
* [X] Meeting with Bertrand Cuissart (2h)
* [ ] Various learning methods from Weka
  * [ ] summarize the results
    * [ ] show why methods are the best
  * [ ] for the best learners
    * [ ] manually test attribute selection methods
    * [ ] implement a test with grid-search for parameter setting
* [ ] Ronan Bureau
  * [ ] put online some new decision trees
    * [ ] for jak2 data
      * [ ] without partial charge
    * [ ] for Mutagenicity data
* [.] RPMP
  * [ ] only extract and filter frequent subgraphs in Mutagenicity data
  * [ ] run DDPMine on those patterns
  * [ ] perform the classification with the remaining patterns
    * [ ] build a binary matrix where descriptors are the found patterns
    * [ ] call any classification method on it (e.g., via Weka)
  * [X] what are the other possibilities?
    * [X] CIMCP
* [ ] General
  * [ ] write dr

2013-02-08 Fri 09:56:59 CET
* [o] Experiments
  * [X] change the scripts so that we give as a parameter a string (or an list of strings) that correspond to the datasets to process
  * [X] disable SVM learning on the data (only weka methods are tested)
  * [o] analyse the results so far and pick the best Weka methods
    * [X] run the regression scripts for each Mohr's dataset separately
    * [X] analyse the results for each dataset (dhfr is missing for now)
      * [X] find the best methods
      * [X] compare with other datasets
* [.] RPMP in Mutagenicity data
  * [X] convert the mined subgraphs into the DDPMine format


2013-02-07 Thu 10:29:13 CET
* [o] Experimental scripts
  * [X] run on Mohr's datasets, with a test sampling rate of 10% and 20% (with the same other settings)
  * [X] run on Jak2 data, with a test sampling rate of 10% and 20% (with the same other settings)
  * [X] run on Jak2 data, with a larger minimum leaf size (30)
  * [X] for classification methods, remove the DTNB algorithm (it is too slow for learning)
* [o] Ronan Bureau
  * [X] How to remove the partial charge from itemsets?
    * [X] in jak2 source code, do not add the partial charge item to the itemsets
* [X] Read document about classification of molecules

2013-02-06 Wed 10:12:08 CET
* [X] Write and send email for reporting the last results (1h)
* [X] Write and send an email to Bertrand Cuissart
* [X] Run experiments on Mutagenesis, with very high minimum leaf sizes

2013-02-05 Tue 09:34:36 CET
* [X] Skype Meeting (Jan Ramon, Bertrand Cuissart) (30 min)
* [ ] Weka usage in learning
  * [ ] How to use weka for the virtual screening problem as well?
    * [ ] should obtain the actual predictions and write them into a file
    * [ ] do some regression on the class labels (i.e., +1 and -1)?
* [ ] New clues for geometrical pattern mining
  * [ ] is it possible to integrate the target-ligand interactions as new items in itemsets?
* [ ] New clues for RPMP
  * [ ] Check the output format of Gaston
  * [ ] Try to use DDPMine
  * [ ] think about the problem of mining closed patterns when there exists an order on the item alphabet
    * [ ] main principle: when a pattern including A has been computed, do not generate any super-patterns containing subgraphs of A

2013-02-04 Mon 09:39:57 CET
* [X] Meeting with Bertrand Cuissart (1h)
* [X] Jak2 code
  * [X] weighting the decisions made by each decision tree in a forest with the size of the used leaf
* [X] Prepare Tuesday's Skype meeting
  * [X] Mohr and Jak2 datasets: run the scripts for computing the performances of each Weka learning method
  * [X] Mutagenicity: run the scripts for computing the classification results
* [X] Mohr datasets
  * [X] perform the prediction with various Weka learners
* [o] General
  * [X] Key Discovery
    * [X] finish implementation of HCA for FAK
      * [X] implement minimality check in candidate generation


2013-02-01 Fri 09:54:01 CET
* [X] Write a meeting report for yesterday's meeting
  * [X] copy the document from netbook
  * [X] write the document
* [X] Mutagenicity
  * [X] run new experiments, by reducing the minimum leaf size

2013-01-31 Thu 09:32:50 CET
* [X] Jak2 code
  * [X] add the searching for occurrences in the leaves
  * [X] Finish the scripts to automatically tests the methods and write the results (for regression and classification)
    * [X] regression
      * [X] read the predictions and write the average values in a csv file
    * [X] classification
      * [X] adapt the script that is supposed to compute the predictions with weka
      * [X] adapt the script that is supposed to summarize the obtained results
  * [X] Mohr's datasets
    * [X] check the preprocessing or run it if necessary
    * [X] copy the data to appo
    * [X] write some scripts to prepare the experiments
    * [X] run some experiments with random forests
* [X] Send an email to Bertrand
* [ ] Mining geometrical patterns
  * [ ] work on size-2 patterns
    * [ ] create the couples of itemsets that can lead to frequent patterns
      * [ ] generate the distances for connecting a couple of vertices
      * [ ] build and testing the corresponding graphs
    * [ ] adding some links to the pattern tree for creating a pattern lattice (with a greedy approach)
  * [ ] Perform experiments with the multi-conformational version of DUD datasets
    * [ ] Random forest size: 50, Test sampling rate: 0.1
  * [ ] Could we also use itemset mining to represent the interactions between target binding site and the ligand?
    * [ ] which representation of the data?
* [X] General
  * [X] Key Discovery
    * [X] for attributes, create a static data structure that stores all the attributes indexed by their local id and their class id

2013-01-30 Wed 09:21:52 CET
* [X] Preparing meeting with Ronan Bureau
  * [X] Run new experiments
    * [X] on Jak2 with decoys
      * [X] check how is the preprocessing
      * [X] copy the data if necessary
    * [X] on DUD with multiconformational version
      * [X] check how is the preprocessing
      * [X] copy the data
  * [X] Mutagenicity
    * [X] Also obtain results with other classifiers
    * [X] Compare those results with referenced results from the Benchmark website
  * [X] Jak2
    * [X] Check how to obtain some tests used with protein-ligand interactions
    * [X] Run one test by learning a decision tree on the whole dataset (no segmentation into learning/test)
  * [X] Documents to prepare/take
    * [X] Molecular descriptors
      * [X] used by Muggleton
      * [X] used by us
    * [X] Mutagenicity results and comparison
      * [X] obtain results with and without molecular descriptors
        * [X] copy molecular descriptors file to the experiment folders
        * [X] ensure the conversion to the SVMLight format
        * [X] compute results manually for a selection of learning approaches
      * [X] compare with given results from http://doc.ml.tu-berlin.de/toxbenchmark/results.html
    * [X] Jak2 results
      * [X] decision trees
      * [X] quantitative results
* [X] Jak2 code
  * [X] Start some scripts to automatically tests the methods and write the results (in regression only)
* [X] General
  * [X] update configurations on machines

2013-01-29 Tue 10:10:57 CET
* [X] Preparing meeting with Ronan Bureau
  * [X] Run new experiments
    * [X] on Jak2 with decoys
      * [X] check how is the preprocessing
      * [X] copy the data if necessary
* [X] Jak2 code
  * [X] Start some scripts to automatically tests the methods and write the results (in regression only)
* [X] General
  * [X] update configurations on machines

2013-01-28 Mon 09:14:49 CET
* [X] Meeting with Bertrand Cuissart (1h30)
* [X] Preparing meeting with Ronan Bureau
  * [X] Mutagenicity
    * [X] Look for results from the literature
* [X] Jak2 code
  * [X] Test a variety of regression methods from Weka to study their general behaviors
* [X] Virtual Screening
  * [X] summarize the last experimental results with and without molecular descriptors
* [X] Send an email to Ronan to plan a meeting together
* [X] General
  * [X] update configurations on machines

2013-01-25 Fri 10:00:29 CET
* [X] Write some scripts to analyse and test the obtained DT in a classical regression context
  * [X] add the molecular descriptors
  * [X] run the svm learning for a regression task
  * [X] analyse the results
    * [X] compute a RMSE value for each possible pattern language
* [X] Jak2 data (with decoys)
  * [X] run the preprocessing of the data in order to compute molecular descriptors
    * [X] for jak2cdockerDecoysSystematicSearch 1 and 2

2013-01-24 Thu 11:04:25 CET
* [X] Jak2 data (with decoys)
  * [X] run the preprocessing of the data in order to compute molecular descriptors
    * [X] for jak2cdockerDecoysSystematicSearch1
    * [X] for jak2cdockerDecoysSystematicSearch2
* [X] Jak2 experiments
  * [X] let the size of random forests vary (only with itemsets and anchor atoms)
    * [X] size=1,   samplingRate=1
    * [X] size=10,  samplingRate=0.2
    * [X] size=50,  samplingRate=0.1
    * [X] size=100, samplingRate=0.05
  * [X] run the experiments
* [X] DUD datasets
  * [X] run the analysis of the whole set of experiments on appo

2013-01-23 Wed 10:28:10 CET
* [X] Jak2 code
  * [X] add to the mutaDataRandomForest example a converter to csv file
* [X] Preparing departure
  * [X] put new version of code on dropbox
  * [X] update gpodder
  * [X] install eclipse on netbook
  * [X] install eclim on netbook
  * [X] update vim config and general config
  * [X] check experiments (jak2, Mutagenesis, and DUD VS result analysis)
    * [X] for analysing the DUD results, block the removing of unused files and segment the process into two steps: learning and testing
    * [X] what should be done for jak2 and Mutagenesis?
      * [X] run the preprocessing of jak2 data with decoys
  * [X] Carte week-end: add a photo

2013-01-22 Tue 08:53:21 CET
* [X] Jak2 code
    * [X] copy the necessary jak2 data
      * [X] preprocess the data on fox
      * [X] compress data
      * [X] move the archives to appo
      * [X] extract the data
* [X] Jak2 data (without any decoys)
  * [X] run the preprocessing of the data in order to compute molecular descriptors
  * [X] write experiment scripts (with and without molecular descriptors)
  * [X] run the experiments
* [X] Run experiments on Mutagenesis datasets with 4D information
  * [X] is the preprocessing OK?
  * [X] write some experiment scripts
  * [X] run the experiments
* [X] DUD datasets
  * [X] run the analysis of all experiments globally
  * [X] run with 32 positive examples instead of 8
* [X] Send an email to Guillaume Poezevara
* [X] General
  * [X] Carte week-end

2013-01-21 Mon 10:58:18 CET
* [X] Jak2 code
  * [X] write a working example containing simple cross validation
  * [X] move the experimental framework to the appo server
    * [X] create a jak2 build folder
    * [X] copy the necessary Mutagenesis data
      * [X] compress data
      * [X] move the archives to appo
      * [X] extract the data
* [X] Run experiments on Mutagenesis datasets with 4D information
  * [X] is the preprocessing OK?
* [X] General
  * [X] starting HCA implementation
    * [X] write attributes and attribute sets classes
  * [X] booking train tickets

2013-01-18 Fri 09:24:22 CET
* [X] DUD experiments (5h)
  * [X] run the testing of the SVM models with and without the integration of molecular descriptors
  * [X] summarize the results
* [X] Skype Meeting with Jan Ramon, Bertrand Cuissart, Bruno Crémilleux (3h30)
  * [X] Molecular descriptors + Patterns
    * [X] list of used molecular descriptors: logp, #heavy atoms, #rotors, #residues, energy, molecular weight, total charge, #HBD, #HBA, dipole moment
    * [X] results (AUROC, BEDROC): 
      * [X] molecular descriptors only:       0.85, 0.39
      * [X] patterns only:                    0.87, 0.57
      * [X] patterns + molecular descriptors: 0.88, 0.48
  * [X] Extending the pattern language
    * [X] several possibilities:
      * [X] emerging conjunctions of fragments
      * [X] frequent closed fragments
  * [X] Jak2
  * [X] Geometrical pattern miner
    * [X] integrated the closed frequent itemset mining to the jak2 code
    * [X] mining size-2 patterns with a pattern miner strategy
  * [X] Future work
    * [X] build a set of molecular descriptors with Ronan Bureau
    * [X] finish experiments for the whole DUD dataset and comparing results with Muggleton

2013-01-17 Thu 10:07:03 CET
* [X] DUD experiments
  * [X] find the last version of experiment analysis scripts
  * [X] select 5 targets
  * [X] copy experiments results to new destination
  * [X] copy the files containing the molecular descriptors
  * [X] run experiment analysis scripts on those results
    * [X] molecular descriptors only
    * [X] molecular descriptors + patterns
    * [X] patterns only
  * [X] run with a very limited amount of data (only k=8)
    * [X] copy the data
    * [X] filter the data
    * [X] run the scripts
* [X] Mining geometrical patterns
  * [X] work on size-2 patterns
    * [X] create the couples of itemsets that can lead to frequent patterns
      * [X] compute the intersection of the extensions for calculating the support
      * [X] adding a patternMining package with options, traits, etc.
      * [X] refining the generation strategy
        * [X] if a couple is unfrequent, do not consider the super-patterns of the second vertex

2013-01-16 Wed 10:26:17 CET
* [X] Run experiments for DUD datasets with molecular descriptors
  * [X] test an independent experiment
  * [X] compile all the experiments
  * [X] run the random forest learners
* [X] Mining geometrical patterns
  * [X] work on size-2 patterns
    * [X] create the couples of itemsets that can lead to frequent patterns
      * [X] loop over the itemsets in a depth-first order
      * [X] compute the intersection of the extensions for calculating the support
* [X] General
  * [X] write dr

2013-01-15 Tue 09:50:57 CET
* [X] Mining geometrical patterns
  * [X] work on size-2 patterns
    * [X] write the skeleton of the pattern miner
* [X] Fixing problems with preprocessed data
  * [X] re-run DUD datasets preprocessing
  * [X] re-run multi-conformation (MC) DUD datasets preprocessing
  * [X] re-run Decoys preprocessing for MC data

2013-01-14 Mon 10:30:18 CET
* [X] Jak2
  * [X] integrate the mined itemsets to the existing refiners
    * [X] reading and creating itemset for each atom (input option)
    * [X] replacing the itemsets in atomItemset refinement option with the dynamically created itemsets
    * [X] testing the new version of itemset mining and refinement
* [X] New candidate generation strategy for geometric patterns
  * [X] work on size-2 patterns
    * [X] build the pattern tree first
* [X] Meeting with Bertrand Cuissart (1h)
  * [X] planning experiments on main current topics
* [X] General
  * [X] write dr

2013-01-11 Fri 09:34:35 CET
* [X] New candidate generation strategy for geometric patterns
  * [X] work on size-2 patterns
    * [X] mining closed frequent itemsets in vertex labels from jak2 code
      * [X] read itemsets mined by LCM
        * [X] create a list of itemsets in main memory
        * [X] create the extension of patterns in main memory
        * [X] compute real support of itemsets
        * [X] keep real extension of frequent closed itemsets
* [X] Meeting with Bertrand Cuissart (2h30)
* [X] General
  * [X] write dr

2013-01-10 Thu 10:49:37 CET
* [X] Preprocesing mutagenesis data
  * [X] fix problems with disk usage on servers
    * [X] compressing results of old experiments
    * [X] removing old experiments folders
  * [X] run preprocessing of DUD 4D data
    * [X] building the merged decoys data
    * [X] building the data for each individual target
    * [X] copy gpm_bench to fox server disk
    * [X] re-run a test with minimum frequency of 10%
* [X] New candidate generation strategy for geometric patterns
  * [X] work on size-2 patterns
    * [X] mining closed frequent itemsets in vertex labels from jak2 code
      * [X] after reading the db, create the input file for LCM
        * [X] keep in memory the transaction ID corresponding to each molecule (with min-max bounds)
      * [X] run LCM
* [X] General
  * [X] write dr

2013-01-09 Wed 09:54:13 CET
* [X] Read emails
* [X] Summarize what has to be done with experiments
* [X] DUD virtual screening
  * [X] preprocess the 4D version of the DUD datasets
    * [X] write and execute a script for building folders for each target
    * [X] copy the obtained data to the preprocessing code
* [X] Mining emerging patterns
  * [X] Discussion with Guillaume Poezevara and compiling the bench source code for quick testing
  * [X] first tests with Mutagenesis data

2012-12-21 Fri 08:48:47 CET
* [ ] Jak2
  * [ ] write experiment scripts for Mutagenesis 4D dataset
  * [ ] DUD
  * [ ] in preprocessing code, add the dipole moment (in 3D)
  * [ ] add the calculation of the dipole moment and add it to the molecular descriptors: x,y,z and length
* [X] Skype meeting (Jan Ramon, Bertrand Cuissart, Bruno Crémilleux, Julien Rabatel) (1h)
* [X] Meeting with Bertrand Cuissart and Bruno Crémilleux (4h)
* [ ] Muggleton's work
  * [ ] create list of molecular descriptors
  * [ ] send the list to Ronan
* [ ] Admin
  * [ ] Send email for deleting unneeded account (ext-rabatel)
* [o] General
  * [ ] print train tickets
  * [X] send email to Rémi

2012-12-20 Thu 08:49:39 CET
* [o] Jak2
  * [X] copy molecular descriptors into experiments datasets
  * [X] run post-processor for learning SVM with molecular descriptors
* [X] Mutagenesis
  * [X] Run preprocessing of multiconformational Mutagenesis dataset
* [o] General
  * [X] write dr

2012-12-19 Wed 09:09:58 CET
* [X] Jak2
  * [X] Send an email to Ronan Bureau (1h)
  * [X] Finish integration of molecular descriptors into SVM learning
    * [X] when the occurrence matrix out of a random forest, also add molecular descriptors
    * [X] test it
      * [X] create a fake molecular_descriptors file
      * [X] conversion of two existing files
  * [X] Redo preprocessing of DUD datasets (uniconformational) to generate molecular descriptors
  * [X] When a SVM model has been learnt, share the weight of apparently equivalent attributes
    * [X] learn an example model
    * [X] check how models look like
    * => support vectors seem to contain all the attributes, with full weight (weight is on the vector, not on the attributes)
* [X] Mutagenesis
  * [X] Run preprocessing of multiconformational Mutagenesis dataset
    * [X] create the input data folders (1 for each run)
    * [X] read data segmentation (learning/test, folds)
    * [X] test it

2012-12-18 Tue 09:04:47 CET
* [X] Preprocessing of 3D data (6h)
  * [X] source code under git
  * [X] add a processor that compute and store molecular descriptors
    * [X] implementing the computation of some molecular descriptors
    * [X] writing the molecular descriptors on file
    * [X] run the preprocessor during the global preprocessing of the input data file
    * [X] testing it
* [X] Jean-Philipe Métivier's seminary on Constraint Programming (1h)

2012-12-17 Mon 08:07:38 CET
* [X] Jak2
  * [X] Run evaluation scripts for results of random forests dedicated to SVM (large size, low test sampling rate)
    * [X] generated files (models, data files) should be removed after used (i.e., all files with '.dat' in the name
  * [X] Check completed results obtained with old configuration of random forests
* [X] Install
  * [X] install dropbox for synchronizing the data files
  * [X] link to the git folders for configuration (vim, ssh, general)
* [X] Admin
  * [X] fill time-sheets for KULeuven
  * [X] fill the forms on Wiki for Migrant project
  * [X] sign, scan and send the time-sheets
* [X] General
  * [X] write dr
  * [X] book train tickets!

2012-12-14 Fri 09:21:58 CET
* [X] Jak2
  * [X] Retrieving results
  * [X] re-run experiment evaluation scripts
  * [X] create summary of results
  * [X] move completed experiments
  * [X] Skype meeting (Bertrand Cuissart, Bruno Crémilleux, Jan Ramon, Julien Rabatel) (1h)
  * [X] Meeting with Bertrand Cuissart and Bruno Crémilleux (1h30)
  * [X] Run new experiments to give the SVM an advantage
    * [X] with large forest size (500)
    * [X] small tests sampling rate (1%)
    * [X] with a small and a large minimum leaf size (2,5,10)
    * [X] generate settings files
    * [X] copy the files
    * [X] compile them

2012-12-13 Thu 10:30:13 CET
* [o] Jak2
  * [X] move old experiments into another folder
  * [.] SVM learning for virtual screening
    * [.] use regression and ranking SVM in addition of classification
      * [o] create the models with the same input data?
        * [X] for ranking: add rank information to SVM input data files
        * [X] test whether it still works with classification
        * [ ] add creating regression model
  * [.] run the interpretation of prediction results and computation of quality measures
    * [ ] for random forests
    * [ ] for SVM
    * [X] parallelize the computation of quality measures
  * [X] add experiment id to csv results file
  * [ ] Check new experimental results

2012-12-12 Wed 10:20:40 CET
* [O] Jak2
  * [X] Answer Leander's email
  * [X] Look at the SVM learner proposed by Leander
  * [X] Preparing data for completing DUD datasets and sending them to Ronan Bureau
  * [X] Learning a SVM on the randopm forest features
    * [X] locally install SVM-light
    * [X] convert occurrence matrices into SVM-light data format
    * [X] perform the learning and the classification
    * [X] read SVM results
    * [X] evaluate the results

2012-12-11 Tue 10:00:34 CET
* [X] Jak2
  * [X] send email to Leander Schietgat about SVM libraries
  * [X] Convert the random forest learned into an occurrence matrix
    * [X] write a specific post-processor
      * [X] read each decision tree in deterministic order
      * [X] for each molecule
        * [X] for each decision tree
          * [X] look for the corresponding leaf (classification step) => '1'
          * [X] for all other leaves, write a '0'
      * [X] write the result as a csv file with ligand name and target value
      * [X] integrate to the virtual screening framework
      * [X] in the experimental framework, check whether resulting csv files are correctly moved to the right place
  * [X] Prepare and run new experiments for DUD datasets
* [X] Literature reading
  * [X] read one paper
* [X] Attending a PhD thesis defense (NLP domain) (1h)
* [X] General
  * [X] write dr
  * [X] book train tickets

2012-12-10 Mon 10:27:00 CET
* [X] Jak2
  * [X] work on update for mips new version
    * [X] reorganize the settings file
    * [X] write a new example with itemset-based atom typing
    * [X] commit and push the current version
    * [X] writing email
* [X] Answer Bruno Cremilleux's email
* [X] send ACM Computing Survey about Sequential pattern mining algorithms
* [X] General
  * [X] write dr

2012-12-07 Fri 09:27:16 CET
* [X] Jak2
  * [X] work on update for mips new version
    * [X] remove all examples that are not working
    * [X] rename working example
* [X] General
  * [X] remove paiement SpiderOak

2012-12-06 Thu 10:41:25 CET
* [X] Jak2
  * [X] work on update for mips new version
    * [X] working example using last mips version

2012-12-05 Wed 10:12:21 CET
* [O] General
  * [ ] write dr
  * [X] book train tickets for end of month
  * [X] print train tickets for today
  * [.] move to netbook
    * [X] some Chemoinformatics papers/lessons 
  * [X] take a few papers
    * [X] Arnaud Soulet's EGC paper
    * [X] Muggleton's papers
    * [X] Data-peeler paper
  * [X] update ipod content
  * [X] copy code from drj
  * [X] to take in luggage
    * [X] book
    * [X] cables for phone/ipod

2012-12-03 Mon 08:12:21 CET
* [X] 'Groupe de Travail Chémoinformatique'
  * [X] finish presentation
    * [X] introduction
    * [X] related work
      * [X] add Muggleton's work
      * [X] add Kernel methods, etc.
    * [X] experiments
    * [X] conclusion and future work

2012-12-02 Sun 12:40:11 CET
* [X] Experiments
  * [X] Mutagenicity

2012-12-01 Sat 14:32:20 CET
* [X] Experiments
  * [X] DUD datasets
    * [X] mine frequent itemsets
    * [X] copy datasets into $build/jak2/data
    * [X] summarize results
      * [X] copy file into local laptop
      * [X] find best settings
    * [X] write experiment scripts
    * [X] generate and copy settings files
    * [X] run compile

2012-11-30 Fri 09:16:58 CET
* [O] Virtual Screening
  * [X] re-add learner log output file to jak2 code
  * [X] create a script to regularly clear all 'learner-output.txt' files
  * [X] re-run experiments
    * [X] remove everything in 'done'
    * [X] copy settings to 'to_preprocess'
    * [X] clear 'waiting_list'
    * [X] run compilation of each setting
    * [X] run scripts
* [ ] Chemoinformatics group meeting
  * [ ] Introduction
  * [ ] Virtual Screening
  * [ ] Geometric patterns
* [X] Mutagenicity experiments
  * [X] check openbabel source code
  * [X] preprocess mutagenecity datasets
  * [X] modify files with itemsets
  * [X] copy files to the experiment folders
  * [ ] generate experiments
    * [ ] write script
* [o] Pre-process DUD datasets with non-hydrogens filter
  * [X] modify openbabel source code
  * [X] run scripts
  * [ ] run itemset extraction
  * [ ] copy files to the experiment folders
* [X] General
  * [X] write dr
  * [X] book train tickets

2012-11-29 Thu 09:08:31 CET
* [O] Jak2
  * [X] Writing/reading data segmentation for repeating experiments (1h30)
  * [X] re-running experiments evaluating the random forest learner settings
  * [o] generate new preprocessed datasets by filtering the Hydrogen atoms
    * [ ] run preprocessing for
      * [ ] DUD datasets
        * [ ] openbabel-based preprocessing (2 times)
        * [ ] itemset mining with low support (5)
* [X] Chemoinformatics group meeting
  * [X] write presentation outline skeleton
  * [X] select slides from previous presentations
* [X] Various tasks
  * [X] Send email to Bertrand Cuissart for planning early-January
* [X] General
  * [X] write dr

2012-11-28 Wed 10:17:02 CET
* [o] Development environment
  * [X] local install of ruby 1.9.3
* [O] Jak2
  * [X] multi-threading for post-processing experimental results (1h)
  * [X] write new experiments with minimum leaf size parameters (2, 4, 8, 12) and run them
  * [.] generate new preprocessed datasets by filtering the Hydrogen atoms
    * [X] updating the preprocessing source code
    * [o] run preprocessing for
      * [X] mutagenesis
        * [X] openbabel-based preprocessing
        * [X] itemset mining
          * [X] 2D, non-hydrogens
          * [X] 4D, non-hydrogens
      * [ ] DUD datasets
        * [ ] openbabel-based preprocessing (2 times)
        * [ ] itemset mining with low support (5)
    * [ ] updating the scripts generating the experiments
    * [ ] copy generated files into experimental code
    * [ ] generating new experiments for
      * [ ] mutagenesis datasets
      * [ ] DUD datasets
* [ ] Mips and Jak2 updating code
  * [ ] test the compiling code
* [ ] Mutagenesis
  * [ ] write experiments (after obtaining first results for the random forest experiments)
  * [ ] run experiments
* [o] General
  * [X] write dr
  * [ ] 1&1: send payment documents and email

2012-11-27 Tue 09:54:56 CET
* [X] Jak2
  * [X] Skype Meeting (Bertrand Cuissart, Jan Ramon) (2h30)
    * [X] meeting preparation (summarizing experimental results)
    * [X] meeting
    * [X] write and send report meeting
* [X] Preparation of 2012-12-04 Chemoinformatics Workshop (5h)
  * [X] writing and sending a summary of presentation
  * [X] retrieving useful documents for the presentation
  * [X] outline for the presentation

2012-11-26 Mon 09:29:07 CET
* [X] Jak2
  * [X] check experimental results (1h)
  * [X] update Friday's results and put them online (30min)
  * [X] run new experiments (2h)
    * [X] evaluating the impact of random forest settings
      * [X] write experiments
      * [X] compile the experiments
      * [X] run the experiments
  * [X] multi-threading for post-processing experimental results (1h)
* [X] Chemoinformatics workshop and project meeting preparation (1h30)
  * [X] outline for presentation
  * [X] meeting with Bertrand Cuissart and Alban Lepailleur
* [X] mips and jak2 code (2h)
  * [X] start the update of jak2 code to fit with mips code
  * [X] mips code up and running
* [X] General
  * [X] Confirm train tickets for Montpellier next week

2012-11-23 Fri 11:18:48 CET
* [X] Geometric pattern mining
  * [X] Adapting the source code to the mips library last version (4h)
    * [X] pull jak2 last version
    * [X] create a new git branch
    * [X] pull mips last version
    * [X] note errors at compile time
    * [X] Skype meeting with Thomas Fannes
    * [X] try to recompile after removing old install
  * [X] retrieve and summarize experimental results and send email to Jan Ramon, Bertrand Cuissart and Bruno Crémilleux with main results (1h30)
* [X] Finishing PAKDD 2013 review (2h30)

2012-11-22 Thu 09:51:21 CET
* [X] Jak2
  * [X] implement and test BEDROC measure for evaluating virtual screening (4h)
* [X] PAKDD 2012 paper review (1h)
  * [X] algorithms
* [X] Mutagenesis
  * [X] preparing experiments on 2D and 4D data
    * [X] preprocessing (3h)
      * [X] 2D
        * [X] converting into correct format
        * [X] extracting frequent closed itemsets for atom-typing
        * [X] copy preprocessed datasets to be used with learner
      * [X] 4D
        * [X] converting into correct format
        * [X] extracting frequent closed itemsets for atom-typing
        * [X] copy preprocessed datasets to be used with learner
* [X] Post-doc project (30min)
  * [X] read and modify Bertrand Cuissart's proposition of summary
  * [X] send modified version

2012-11-21 Wed 09:21:08 CET
* [X] Jak2
  * [X] implement the scripted creation of a summary over all the experiments (4h)
  * [X] write new experiments for evaluating a 2-step refiner
* [X] PAKDD 2012 paper review (1h)
  * [X] finish reading
    * [X] definitions and formalism
  * [X] reading review form
* [X] Post-doc project
  * [X] read and modify Bertrand Cuissart's proposition of summary
    * [X] reading and finding spelling mistakes
  * [X] send email to Guillaume Poezevara to schedule a Skype meeting
* [X] Skypatterns and 3D molecular data (1h)
  * [X] look at how skypatterns have been used in the past with 2D molecular data
  * [X] propose 3D information-based measures to use for sky-pattern mining
  * [X] Discussion with Bruno Crémilleux about the definition of 3D-based measures for molecules
* [X] General
  * [X] book train tickets for Montpellier (5/12 -> 8/12)
  * [X] option on booking train tickets for Montpellier (Christmas)

2012-11-20 Tue 08:21:37 CET
* [X] Jak2
  * [X] Skype Meeting (Jan Ramon, Bertrand Cuissart) (1h30)
    * [X] discussion about DUD benchmark data and DT learner for virtual screening
  * [X] adapting evaluation measures to bags obtained with random forests (2h)
  * [X] fixing problems with connection to servers for experiments (1h)
* [X] CODAG group meeting: discussion about Constraint Programming and Itemset Mining (2h)
* [X] Meeting with Bertrand Cuissart (1h30)
  * [X] point about the different project aspects and progress

2012-11-19 Mon 09:16:30 CET
* [o] Jak2
  * [X] run the tests for DUD and virtual screening (2h)
    * [X] compiling all the tests
    * [X] running the tests
    * [X] check whether the settings files are fine
  * [o] defining a richer refinement operator (adding a new edge AND a new atom-typing label)
    * [X] implement it
    * [ ] test it
  * [ ] testing the balance of DT again
  * [X] writing ruby scripts to compute quality measures for virtual screening 
    * [X] find first version of scripts
    * [X] check whether the prediction results reading works correctly
* [X] Mutagenesis
  * [X] download and start preprocessing of multi-conformational mutagenesis data
  * [X] answer Ronan's email

2012-11-16 Fri 09:53:22 CET
* [X] Jak2
  * [X] Virtual Screening
    * [X] DUD dataset: check the data segmentation provided for experiments (2h)
    * [X] add and test a constraint of maximum increment size in the itemsets used to label the atoms (2h)
    * [X] fixing problem with distance refiner: no distance-based candidates are generated in the experiments (1h30)
    * [X] write scripts for tests (1h)
* [X] Meeting with Bruno Crémilleux (30min)
* [X] Meeting with Winn Voravuthikunchai (30min)

2012-11-15 Thu 09:37:32 CET
* [X] Jak2
  * [X] precisely measure whether decision trees obtained via pattern-based atom-typing are balanced (2h)
    * [X] results: atom-typing itemsets segment the data in  45%/55% splits in average
  * [X] Virtual Screening (4h30)
    * [X] build and copy the data containing the union of all the decoys from all targets in DUD data
    * [X] test the segmentation of data
    * [X] run one experiment to test the framework feasability
* [X] Post-doc project
  * [X] Meeting with Guillaume Poezevara (about presentation preparation for the 04/12 and R integration in a C++ project) (1h30)
* [X] General
  * [X] write dr
  * [X] train tickets (5/12 --> 9/12)

2012-11-14 Wed 09:35:34 CET
* [X] Jak2
  * [X] Virtual Screening (6h)
    * [X] in jak2 code, creating the learning / test datasets
* [X] General
  * [X] write dr
  * [X] answer to Sandra's email

2012-11-13 Tue 09:36:43 CET
* [o] Jak2
  * [X] Sype meeting (Jan Ramon, Bertrand Cuissart) (2h)
    * [X] Pattern-based atom-typing
    * [X] DUD benchmarking
      * [X] no results yet
      * [X] Ronan will prepare a set of conformations for each target
    * [X] send meeting report
  * [X] checking whether decision trees obtained with pattern-based atom typing are unbalanced or not (1h)
* [X] François Rioult's seminar + CODAG group meeting (2h)
* [X] General
  * [X] write dr
  * [X] check Spain for holidays
  * [X] book train tickets for next week

2012-11-12 Mon 15:14:30 CET
* [X] Meeting with Ronan Bureau (CERMN) (2h30)
* [X] Prepare and send DUD data to Ronan for multi-conformation generation for every target (1h)
* [X] Continue preparation of DUD datasets and Virtual Screening (2h)
  * [X] check whether all preprocessing has been done

2012-11-09 Fri 09:35:36 CET
* [.] Virtual Screening
  * [o] Preparing the dataset
    * [X] preprocess each DUD dataset individually
    * [X] keeping only 1 conformation for each molecule in the decoys (other conformations actually are duplicates)
    * [ ] for each target, build the learning datasets and corresponding test datasets
* [ ] Jak2 experiments
  * [ ] preprocessing last data sent by Ronan (cdocker + decoys + systematic conformation search)
    * [ ] mining the atom typing itemsets in the preprocessed datasets
  * [ ] run tests on laboratory servers
    * [ ] run the tests
* [o] Post-doc project
  * [X] Mail to Ronan Bureau (about the benchmark dataset for mutagenesis)
  * [o] Benchmark dataset
    * [X] trying a first preprocessing on sdf data (with 2D coordinates)
    * [X] mining atom-typing itemsets
    * [ ] testing the decision tree learner on it
* [X] General
  * [X] write dr
  * [X] sent 1&1 email with documents for paiment


2012-11-08 Thu 09:22:12 CET
* [X] Jak2 experiments
  * [X] preprocessing last data sent by Ronan (cdocker + decoys + systematic conformation search) (2h30)
    * [X] running preprocessing
      * [X] fixing problems when reading input files
      * [X] reinstalling openbabel on servers
* [X] Pattern-based atom typing (5h30)
  * [X] implement multiple refinements for atom typing
    * [X] when reading the pattern list, create for each pattern a list of pointers to its specializations
    * [X] itemset-based vertex labels are replaced by pointers to one itemset in the previously built pattern list
    * [X] when a vertex is already labeled with atom typing, new candidates are specializations of the existing label itemset
* [X] Post-doc project (30min)
  * [X] Meeting with Guillaume Poezevara (preparing the next project meeting)
* [X] General
  * [X] write dr

2012-11-07 Wed 09:52:14 CET
* [X] Jak2
  * [X] Improving the implementation: printing understandable values of atom typing, instead of item IDs (1h)
  * [X] Jak2 experiments
    * [X] preprocessing last data sent by Ronan (cdocker + decoys + systematic conformation search) (3h)
      * [X] moving all preprocessing code and libraries to servers (for large scale experiments)
    * [X] run tests on laboratory servers
      * [X] write the scripts for preparing the tests (1h)
* [X] Post-doc project
  * [X] Meeting with Guillaume Poezevara (2h)

2012-11-06 Tue 09:01:55 CET
* [X] Jak2
  * [X] Preparing the Skype meeting (1h)
  * [X] Skype meeting Jan, Bertrand and Bruno (1h)
    * [X] write and send the meeting report
  * [X] Short post-skype discussion with Bertrand and Bruno (30min)
  * [X] Pattern-based atom typing
    * [X] collected and summarized experimental results on 4 Mohr's datasets (1h)
  * [X] Meeting with Bertrand (2h30)
    * [X] Atom pairs and line graphs
  * [X] Jak2 experiments
    * [X] preprocessing last data sent by Ronan (cdocker + decoys + systematic conformation search) (1h30)
* [X] Post-Doc Project
  * [X] mail to Guillaume
  * [X] Planning a meeting with Ronan 
    * [X] answer Ronan's email (30 min)

2012-11-05 Mon 09:33:12 CET
* [X] Jak2
  * [X] Virtual Screening
    * [X] preprocess data for virtual screening experimental framework from Muggleton et al. (3h)
      * [X] retrieve the set of negative examples from ZINC and integrate them
    * [X] Atom Typing based on frequent itemsets (4h)
      * [X] write the experiment scripts
      * [X] run systematic experiments with several minimum leaf sizes on Mohr datasets
* [X] General
  * [X] write dr
  * [X] payment for Les Doges
  * [X] payment for 1&1

2012-11-02 Fri 09:50:36 CET
* [.] Jak2
  * [.] Virtual Screening
    * [ ] pre-processing the DUD datasets
      * [ ] extract frequent patterns for atom typing in 3 datasets cox2, trypsin, pdgfrb
      * [ ] extract frequent patterns for other datasets
    * [.] run experiments on datasets
      * [X] list datasets to use for first experiments
      * [ ] write tests for the datasets
        * [ ] how to determine correct leaf size? --> 20
        * [ ] only with random forests to obtain an ordering of molecules
          * [ ] forest size: 100
        * [ ] with pattern-based atom typing
        * [ ] with/without bonds? -> firstly without
  * [ ] General refinement strategy
    * [ ] write different refiners
      * [ ] a refiner with pattern-based atom typing and distance 
  * [ ] Optimization in code
    * [ ] when matching atom typing itemset, if one itemset is not supported by the conformation, return false for the entire ligand
  * [ ] Send email about the experimental framework
    * [ ] Problem definition in the reference paper
      * [ ] LBVS
    * [ ] Evaluation methods
      * [ ] quality measures
    * [ ] How to adapt the DT learner for this problem definition?
      * [ ] Considering the query ligand
        * [ ] Similarity measure
      * [ ] Changes in DT heuristic?
  * [ ] Send email to Ronan
  * [ ] Literature
    * [ ] summary of virtual screening and active learning
    * [ ] summary of activity prediction and toxicity

2012-10-31 Thu 10:09:05 CET
* [X] Jak2
  * [X] Virtual screening
    * [X] finish preprocessing of DUD
      * [X] do and check the conversion of files mol2 -> sdf via OpenBabel
      * [X] read sdf files and assign correct target values
      * [X] the molecule name for decoys is not unique! --> use an id instead corresponding to the index in source file
      * [X] script for preprocessing each of the DUD datasets
      * [X] copy preprocessed data into fox

2012-10-30 Tue 09:09:49 CET
* [ ] Jak2
  * [ ] Skype
    * [ ] progress on DUD datasets
    * [ ] New atom typing refinement strategy
      * [ ] old version: problem with non-sufficient steps to lead to interesting test patterns, a lot of prunable candidates generated
      * [ ] integrated a frequent closed pattern mining step in preprocessing: mining sets of atom properties that are both frequent and closed
        * [ ] interests: limited number of possibilities (around 350/400 patterns), some of those patterns are chemically significant
        * [ ] results: systematically improved results in prediction / variance reduction even with a 1-step refiner, gain in processing time
        * [ ] drawbacks: in current version, an atom can receive only one atom typing label

2012-10-26 Fri 11:28:13 CEST
* [.] Jak2
  * [X] send meeting report
    * [X] discussed
    * [X] to do
    * [X] paper to attach
  * [ ] preprocess DUD datasets
    * [ ] choose some datasets
    * [ ] compute the union of active compounds and decoys
    * [ ] run the openbabel preprocessing of resulting dataset
  * [ ] preprocess new JAK2 datasets
* [X] General
  * [X] install mails
    * [X] unicaen
    * [X] gmail
    * [X] kuleuven
    * [X] lirmm
  * [ ] install oh my zsh

2012-10-25 Thu 08:30:05 CEST
* [ ] Jak2
  * [ ] applications for the DT learner?
    * [ ] protein-ligand binding site discovery
    * [ ] Virtual Screening

* Skype 14h 
  * [ ] data for comparison with DT learner in VS
    * [ ] Data from Ronan's data --> just received them with decoys and multi-conformations
    * [ ] Benchmark of active compounds and decoys DUD
      * [ ] 40 targets with:
        * [ ] active compounds and decoys
        * [ ] 3D coordinates
        * [ ] crystallographic conformation for the ligand! (except for a few targets)
      * [ ] One paper JCIM2010 with results for those data on several approaches (2D, 3D, very limited in 4D)
        * [ ] "A comprehensive comparison of ligand-based virtual screening tools against the DUD dataset reveals limitations of current 3D methods", JCIM2010
      * [ ] Currently building and preprocessing the whole dataset
      * [ ] Implementing the same quality measures
      * [ ] Problem definition:
        * [ ] one is the framework that we currently use: building DT on the whole learning dataset and testing the resulting tree with test dataset --> directly feasible
        * [ ] other possibility is to integrate a ligand query --> from crystallographic ligand pose
          * [ ] with DT: build DT on learning set, retrieve leaf for query, consider all the molecules from same leaf
          * [ ] with RF: build RF; similarity measure with query is the fraction of trees in the forest where the molecule is in the same leaf as the query node
          * [ ] Remark: the DT learning could be modified: we only need the branch for the query ligand --> faster tree construction
          * [ ] how to split the datasets into learning / test data? the test dataset should be sufficiently large

    * [ ] To do 
      * [ ] active learning settings
        * [ ] what has been done in the literature?
        * [ ] how to adapt it to our approach?
        * [ ] 1. first screen a small part of the db
        * [ ] 2. perform the tests
      * [ ] toxicity and activity
        * [ ] work done to combine two problems?
        * [ ] try to find whether toxicity plays a role in the construction of inactive decoys (can a decoy be both active and toxic?)

  * [ ] Next skype meeting, 30/10/2012 at 10am

  * [ ] implement one or two quality measures for VS
  * [ ] fix problem in printing random forest results
  * [ ] analyse last results
  * [ ] run experiments with jak2cdocker dataset (before multi conformations)
  * [ ] run experiments with jak2goldMultiConformations--all
  * [ ] Skype 14h (Jan, Bertran, Bruno)
    * [ ] new version of atom typing
    * [ ] virtual screening evaluation
      * [ ] datasets from DUD
        * [ ] intersection with Mohr's dataset
        * [ ] currently building the datasets and preprocessing them (by adding the decoys)
      * [ ] problem with multi-conformations
        * [ ] ask Ronan to generate conformations for them?
      * [ ] problem with block effect of DT
    * [ ] data from Ronan

2012-10-24 Wed 09:24:19 CEST
* [.] Jak2
  * [X] test atom itemsets on Mohr's data
    * [X] preprocess Mohr's data
      * [X] copy to ruby mining scripts
      * [X] extract patterns for each dataset
      * [X] copy resulting files into Mohr's datasets
      * [X] copy preprocessed datasets into build/data directory
    * [X] adapt ruby scripts for experiments
      * [X] copy them from KULeuven to /home/personnels/rabatel/...
      * [X] write the correct scripts for
        * [X] Mohr data
        * [X] jak2
      * [X] copy the obtained file settings to the to_preprocess folder
    * [X] run DT learner
      * [X] with pattern-based atom typing
      * [X] with classical atom typing
  * [X] run jak2
    * [X] with pattern-based atom typing
    * [X] with classical atom typing
  * [ ] use a sampling method on ligands?
  * [ ] implement a virtual screening quality measure in addition
  * [ ] find benchmark approaches for 3D virtual screening and corresponding data
  * [ ] for large-scale experiments
    * [ ] see group list in the FAQ
* [X] General
  * [X] eee pc
    * [X] install ruby and RoR
    * [X] install ssh server
  * [ ] Karim's birthday

2012-10-23 Tue 08:30:15 CEST
* [X] Jak2
  * [X] finish implementation of itemset labels (2h)
* [ ] Post-Doc Project
* [o] General
  * [X] write dr
  * [.] Rails
    * [X] taggable generator
    * [ ] Railscast
      * [ ] Comments
        * [ ] Convert into generator
      * [ ] YAML config
      * [ ] jQuery File Upload
      * [ ] Cropping images
      * [ ] Token Fields (revised)
      * [ ] Calendars (revised)
      * [ ] ActiveRecord Reputation System
    * [ ] likeable generator
      * [ ] partials
        * [ ] like link
        * [ ] unlike link
        * [ ] ajax
    * [ ] followable/favouritable generator
      * [ ] partials
        * [ ] follow form
        * [ ] unfollow form
        * [ ] ajax
    * [ ] categorizable generator

2012-10-22 Mon 09:02:10 CEST
* [.] Jak2
  * [X] obtain a working source code
    * [X] delete test pre-generator
  * [X] install on /home/personnels/rabatel/
    * [X] install code
    * [X] build it
    * [X] copy data
  * [X] implement new label type with frequent itemsets
    * [X] read itemset file and occurrences
    * [X] build an atom option with corresponding itemset
      * [X] with script: write on file mapping property->itemId
      * [X] read mapping of items
      * [X] convert one atom into one itemset
  * [ ] think about mining pairs of atoms
  * [ ] test data from Ronan with current version
  * [ ] list references about atom pairs
  * [ ] run experiments
  * [ ] find benchmark data for virtual screening
  * [ ] Find datasets for evaluation
    * [ ] in multi-conformations: data from paper ...
* [ ] Post-Doc Project
  * [ ] cahiers de labo
  * [ ] study the Chi² test interestingness measure for patterns
  * [ ] read documentation about statistical tests
* [O] General
  * [X] write dr
  * [X] asus
  * [X] SpiderOak
    * [X] pay last bill
    * [X] terminate contract
  * [.] Rails
    * [X] taggable generator
    * [ ] likeable generator
      * [ ] partials
        * [ ] like link
        * [ ] unlike link
        * [ ] ajax
    * [ ] followable/favouritable generator
      * [ ] partials
        * [ ] follow form
        * [ ] unfollow form
        * [ ] ajax
    * [ ] categorizable generator

2012-10-19 Fri 09:42:57 CEST
* [X] Jak2
  * [X] write the current situation (2h)
    * [X] source code update and installation on local workspace
  * [X] Ruby scripts to convert the representations of 3D information into atom pairs (3h)
    * [X] read protein data
    * [X] read target values
    * [X] generate different itemset representations
      * [X] atoms and their labels
    * [X] mine frequent patterns (or frequent closed patterns)
    * [X] count multi-instance support of atoms
      * [X] count support
      * [X] store list of occurrences (list of ligands)
    * [X] write obtained patterns and their occurrences
    * [X] closed frequent atom pairs (CFAP)
      * [X] write geometric discretizer
* [X] General
  * [X] download Coursera:
    * [X] algorithms

2012-10-18 Thu 09:48:32 CEST
* [.] Post-Doc Project
  * [X] read documentation provided by Jérémie Le Goff about the Ames test for mutagenecity and cancerogenicity of molecules (1h)
    * link: /export/home/rabatel/Documents/Bibliography/ExpertDomainDocumentation/Ames test 10_2012.docx
    * [X] answer Jérémie's email
* [X] Jak2
  * [X] write the current situation
    * available data:
      * multi-conformations with CDocker
      * multi-conformations with Gold (but no way to have a score for one conformation!)
    * to test:
      * Gold data (on the whole dataset!)
      * CDocker data on the topX conformations
      * also test with random forests
    * analyze results:
      * gain compared with old data
    * implement a virtual screening evaluation measure
    * find benchmard datasets to compare with
  * [X] experiments (2h)
    * [X] find last data sent by Ronan
    * [X] preprocess these data if necessary
* [X] General
  * [X] write dr
  * [X] railscast
  * [X] send sms Karim
  * [X] confirm train tickets

2012-10-17 Wed 12:09:51 CEST
* [o] Post-Doc Project
  * [X] meeting with Bertrand (2h)
* [X] General
  * [X] book train tickets
  * [X] send information form to Arielle
    * [X] fill the form
    * [X] send

2012-10-16 Tue 09:32:38 CEST
* [X] Post-Doc Project
  * [X] Chemoinformatics seminar (Willy Ugarte) (1h)
  * [X] project starting meeting (Jeremie, Bertrand, Ronan, Alban, Bruno, Julien) (1H30)
    * [X] read project
    * [X] list possible future work
      * [X] mining patterns with a continuous target attribute
      * [X] mine conjunctions of patterns with other measures than growth rate
        * [X] information gain, etc.
        * [X] variance reduction
      * [X] improve the results of toxicology prediction
        * [X] prediction/classification based on rules (not only one jumping pattern)
        * [X] statistical properties of such jumping patterns
      * [X] study the problem of using a more elaborate representation of atoms (with disjunctions?)
      * [X] pattern stimulation in other contexts than Bayes?
      * [X] inhibitor mining
        * [X] mining algorithm based on support-based definition of inhibitors
          * [X] define different types of inhibitors
          * [X] propose algorithms to mine them
        * [X] definition based on stimulation?
      * [X] combining QSAR + toxicophores
      * [X] problems related to the super-imposition of patterns
        * [X] introducing a notion of distance between 2 fragments of a pattern
        * [X] need for extracting emerging patterns with a variable growth-rate (or other measures) given the coverage --> see approaches with branch-and-bound search (Direct Discriminative Pattern Mining for Effective Classification)
        * [X] mine the correlation between the toxicity (or other target attribute) and the number of occurrences of a fragment
      * [X] Focusing on the different types of toxicity
        * [X] contextual pattern extraction? --> what is common/different from different types of toxicities?
        * [X] knowledge extraction: rather than only considering the high/low impact on toxicity of a toxicophore, study whether or not its impact is more or less general (power of generatlization of a toxicophore)
* [X] PatternStimulation
  * [X] submit paper
  * [X] send sources to Bertrand and Ryan
* [o] General
  * [X] send email to Radu
  * [X] answer Ryan's email

2012-10-14 Sun 10:44:45 CEST
* [.] PatternStimulation
  * [O] Code
    * [X] implement pattern selection on GR
      * [X] interval computation
      * [X] building the list of patterns to use
    * [X] check what happens with degenerate odds
    * [X] oblige p,q to be frequent as an option
    * [X] measure traditional vs alternative classification
    * [ ] make a test with considering the median value for tests that got rejected by the tests
    * [X] implement naive bayes classification
      * [X] select patterns of size one only
      * [X] patterns to use are all the patterns of size 1! (seems to be implicit with previous)
    * [X] write scripts to 
      * [X] run lots of experiments
      * [X] read and select the best results
      * [X] not running an experiment if it has already be done
    * [X] add to the log names
      * [X] dataset
      * [X] classifier type
      * [X] minFreq
      * [X] minInt
      * [X] N
      * [X] R
      * [X] percentile
* [ ] write dr

2012-10-13 Sat 11:49:50 CEST
* [X] PatternStimulation
  * [X] Code
    * [X] check the median retrieval for slope values
    * [X] compute other measures for flagging
      * [X] features of a flag
        * [X] up or down compared to the reference value
        * [X] ? - how to consider the flag of the median?
      * [X] number of flags that are in the opposed way compared to the selected one

2012-10-11 Thu 07:16:29 CEST
* [.] PatternStimulation
  * [o] Code
    * [X] Test heart dataset with old and new rules
      * [X] look at the global accuracy
    * [X] In case of 10th percentile, never take the first value
    * [o] store information about flags
      * [X] only first flags
      * [X] all flags
      * [ ] compute TN, TP, FP, FN
    * [ ] compute and store success rate (TP, ..) for each fold and each dataset
    * [X] filter for frequency and GR in process
      * [X] frequency in pattern list
      * [X] interest measure in pattern lattice
  * [ ] Extract frequent patterns with low minimum support (1% when feasible?)
  * [ ] Run tests on other datasets
* [ ] General
  * [ ] write dr

2012-10-10 Wed 08:33:21 CEST
* [o] PatternStimulation
  * [o] Code
    * [X] Compute frequencies when needed
    * [X] derivation rules with degenerate odds
    * [X] multi-class classification
    * [X] embedded tests for R
    * [X] logging results
    * [X] change statistical tests!
      * [X] to two sided
      * [X] by considering the differences
    * [X] stimset statistic
      * [X] propose 2 options
        * [X] stimset statistic is the 90th percentile
        * [X] stimset statistic is the reversed 90th percentile (but never the smallest)
  * [X] Store and log:
    * [X] first flag (ask Bertrand about how to say that a flag is up or down)
    * [X] list of flags
    * [X] normal and alternative classifier results
    * [X] n-way classification
  * [X] Skype Ryan 

2012-10-09 Tue 07:58:18 CEST
* [X] PatternStimulation
  * [X] Retrieve datasets from UCI
    * [X] list available data from François's website
    * [X] read them with ruby scripts and write on files
  * [X] General questions
    * [X] how to state whether a flag is up or down?
    * [X] What to do with patterns in interval that are not frequent patterns? (we do not know their support)?
* [X] General
  * [X] write dr

2012-10-05 ven. 14:42:25 CEST
* [X] PatternStimulation
  * [X] Code
    * [X] flagging
      * [X] compute stimset statistic
      * [X] propose a method to check whether a pattern is closed in pattern lattice
      * [X] integrate flagging to classification?
    * [X] Classification
      * [X] what about the uncovered attributes? check in the EP-based classifier
      * [X] is everything working?
    * [X] Ruby scripts
      * [X] extract frequent patterns instead of closed patterns
* [X] General
  * [X] write dr

2012-10-04 jeu. 09:52:46 CEST
* [X] PatternStimulation
  * [X] Writing
    * [X] compile latex sources
  * [X] Code
    * [X] Prepare classification methods
      * [X] cross validation
      * [X] compute probability for each class
* [X] General
  * [X] write dr
  * [X] bank: changing address and withdrawing cash


2012-10-03 Wed 09:11:16 CEST
* [X] General
  * [X] take address justification

2012-10-02 Tue 08:33:06 CEST
* [X] PatternStimulation
  * [X] Code
    * [X] give a numerical id to each pattern lattice node (to allow the creation of a vector) to mark them during a traversal
    * [X] compute the interval of 2 patterns in pattern lattice _interval(p, p')_ -> list<NodePtrContainer>
    * [X] compute the stimset of q on an interval _stimset(q, interval)_ -> list<double>
  * [X] Skype with Ryan
* [X] Jak2
  * [X] skype meeting
* [X] General
  * [X] write dr
  * [X] print train tickets

2012-10-01 Mon 16:12:13 CEST
* [X] PatternStimulation
  * [X] implement the construction of the list of patterns to use for computing the product approximation
    * [X] for emerging patterns
      * [X] ordered by decreasing strength
      * [X] ordered by increasing length
      * [X] ordered by increasing size of increment

2012-09-28 Fri 08:15:20 CEST
* [X] PatternStimulation
  * [X] implement the construction of the list of patterns that are included in the test object
* [X] General
  * [X] write dr

2012-09-27 Thu 08:06:20 CEST
* [X] Jak2
  * [X] run new tests (one for each new dataset)
  * [X] preprocess gold data
* [X] PatternStimulation
  * [X] implement some operations on the pattern lattice
    * [X] keep patterns with a minimum value on given measure
    * [X] select minimal patterns in pattern lattice: children of root
    * [X] select maximal patterns in pattern lattice: leaves
    * [X] compute pattern odds for each class
* [X] General
  * [X] write dr

2012-09-26 Wed 08:55:23 CEST
* [X] Jak2
  * [X] create new versions of the dataset
    * [X] problem: the conformations are not ordered w.r.t. their conformation score
* [X] PatternStimulation
  * [X] in pattern tree, add other parent-to-child relations (i.e., every inclusion relation in the transitive reduction)
  * [X] compute measures
    * [X] frequency
    * [X] growth-rates
    * [X] strength
    * [X] chi²
* [X] General
  * [X] write dr

2012-09-25 Tue 08:55:37 CEST
  * [X] find and save Springer digraph book
  * [X] read paper on slope-one predictors
* [X] General
  * [X] write dr

2012-09-24 Mon 09:46:44 CEST
* [X] Jak2
  * [X] pre-process new data
* [X] PatternStimulation
  * [X] change ruby scripts so that 
    * [X] one datafile is created for each class
  * [X] compute supports and interestingness measures
    * [X] support in each class
  * [X] build pattern tree
  * [X] General
    * [X] write dr
    * [X] book train tickets

2012-09-21 Fri 09:36:42 CEST
* [X] PatternStimulation
  * [X] create C++ project and cmake
  * [X] read patterns from file
* [X] General
  * [X] write dr
  * [X] check credit card
  * [X] files to transfer:

2012-09-20 Wed 07:58:30 CEST
* [X] Jak2
  * [X] send email to Ronan
  * [X] read papers about virtual screening
* [X] Pattern Stimulation
  * [X] write ruby script to:
    * [X] create the mapping value --> item_id
    * [X] convert a dataset into a numerical format
    * [X] add class items (it should be the first in the numerical order!)
    * [X] split the data into 10 folds (for future cross-validation) (feasible with weka?)
  * [X] have a look at Ryan's implementation
* [X] KeyDiscovery
  * [X] look at Gordian alg
  * [X] is it possible to use it as a preprocessor in HCA? --> yes, but we should generate the sampled data...
  * [X] send email to Madalina : no need to ask for implementation
  * [X] how multi-values are considered by François?
* [X] General
  * [X] move files to iPod
  * [X] check messages from SFR
  * [X] File transfers:
    * [X] vim config
    * [X] general config

2012-09-19 Wed 07:58:30 CEST
* [X] Jak2
  * [X] print a selection of papers about virtual screening
  * [X] start implementation of test pre-generator
  * [X] write settings for atom typing and bond multi-level refiners
* [X] Pattern Stimulation
  * [X] download CLOSET+
  * [X] start a new C++ project
    * [X] problem: for emerging patterns, we need patterns that are unfrequent in one class. We could mine frequent patterns in the whole dataset and then calculate the support of each pattern in both classes.
  * [X] Find the slope one paper, SDM 2005
* [X] Key discovery
  * [X] Java project: write architecture
* [X] General
  * [X] send email to Bertrand
  * [X] write dr
  * [X] sfr
  * [X] google maps

2012-09-18 Tue 08:14:08 CEST
* [X] Jak2
  * [X] Meeting
    * [X] print results
    * [X] summarize results with random forests
  * [X] write and send report
* [X] General
  * [X] write dr

2012-09-17 Mon 09:21:56 CEST
* [X] Jak2
  * [X] Summarize results
  * [X] Run new tests
    * [X] random forests
  * [X] Multi-level atom-typing refiner
    * [X] write a multi-level atom-typing refiner (for 2 and 3 levels)
    * [X] test it
  * [X] send e-mail to Bertrand
  * [X] read results from random forests
  * [X] add new settings
    * [X] 1, 2, 3 level bond refiners
    * [X] 1, 2, 3 level atom typing refiners
* [X] General
  * [X] write dr
  * [X] print last bill from SFR
  * [X] install vim plugin to convert from/into CamelCase/SnakeCase conventions

2012-09-14 Fri 08:42:07 CEST
 * [X] Jak2
   * [X] create new experiments
   * [X] summarize results
     * [X] read settings file
     * [X] read cross-validation logs
     * [X] create a csv file given all the experiments
 * [X] Pattern Stimulation
   * [X] read last part of the paper
   * [X] find a closed itemset miner
 * [X] General
   * [X] write dr

2012-09-13 Thu 09:10:25 CEST
* [X] Jak2
  * [X] push new version of jak2
  * [X] push new version of mips
  * [X] update mips remote branch (warning: in the local version of mips, I removed the trypsin example => checkout)
  * [X] pull jak2 and mips before running experiments in KULeuven
  * [X] work on bond refiner
  * [X] generate the experiments settings
  * [X] experiments
    * [X] test new bond refiners on Mohr's datasets
    * [X] redo experiments on random forests (the decision strategy was wrong in the previous ones)
  * [X] Key discovery
    * [X] check implementation of François
    * [X] see what can be taken
* [X] General
  * [X] scan papers for residence
  * [X] write dr
  * [X] RoR
    * [X] save railscast on Test

2012-09-12 Wed 09:09:58 CEST
* [X] Jak2
  * [X] answer Bruno's email
  * [X] add new bond-refiners: one for each atom typing label
  * [X] re-generate data for Mohr's datasets
  * [X] write ruby script to generate settings for experiments
* [X] KD
  * [X] structure of the code
  * [X] read paper HCA
* [X] General
  * [X] call m
  * [X] write dr
  * [X] take train tickets

2012-09-11 mar. 08:33:07 CEST
* [X] Jak2
  * [X] Have openbabel_preprocessing running on local machine
  * [X] Run tests with Mohr's datasets and 2-step refiner
  * [X] Re-run with random forests
    * [X] Mutagenesis (warning: should display the classification results)
    * [X] Mohr's datasets
  * [X] Send meeting report with Ronan
  * [X] Write skype meeting report
* [X] Stimulation of patterns
  * [X] ask Bertrand for Ryan's implementation
* [X] Install jak2 on KULeuven disk space
* [X] Install mips on KULeuven disk space
* [X] Install openbabel on KULeuven disk space
* [X] Install openbabel_preprocessing  on KULeuven disk space
* [X] Install tmux on KULeuven disk space
* [X] General
  * [X] write dr
  * [X] print Madalina's work
  * [X] add global variables to configuration
    * [X] $ruby
    * [X] $java

2012-09-10 lun. 07:55:58 CEST
* [X] Jak2
  * [X] Experiments
    * [X] tests on mutagenesis data (simple patterns and their real support)
    * [X] run tests on other datasets (Mohr et al.)
    * [X] run tests on Jak2
    * [X] start writing script in ruby for experiments (instead of bash)
* [X] Toxicology
  * [X] finish reading paper on pattern stimulation
* [X] General
  * [X] write dr
  * [X] Anniversaire Rémi

2012-09-07 ven. 08:51:41 CEST
* [X] Jak2
  * [X] copy data from kuleuven
  * [X] copy datasets in the jak2 build directory
  * [X] copy openbabel
    * [X] source
    * [X] build
  * [X] write and send meeting report
  * [X] debug the edge and vertex refiner
  * [X] before leaving: install mips and jak2 on laptop!!!
* [X] General
  * [X] write dr
  * [X] add -h to ll alias
  * [X] add git config to general_config
  * [X] print bus timetables for the sea

2012-09-06 jeu. 13:31:22 CEST
* [X] GREYC installation
  * [X] install vim config
  * [X] install ssh config
  * [X] install tmux
  * [X] install mips code
    * [X] source
    * [X] build for mips
  * [X] install jak2 code
    * [X] source
    * [X] build for jak2
* [X] Jak2
  * [X] meeting with Ronan

2012-08-30 Thu 11:57:19 CEST
* [X] Jak2
  * [X] have lumo and logp tests working
  * [X] debug the new refinement type

28/08/2012
* [o] Jak2
  * [X] send xls file with background knowledge
  * [ ] implement a new version of the bond-refiners
    * [ ] a test is a combination of edge and a new vertex label
  * [ ] start implementation of new test generator
    * [ ] generate individual vertices
      * [ ] generate extension
      * [ ] keep frequent ones
  * [X] Skype meeting
    * [X] summary
      * [X] report sent on Friday
      * [X] added the bond type
    * [X] report
    * [X] answer Bruno's email
  * [o] General
    * [X] write dr
    * [ ] call residence in Caen

2012-08-27 Mon 09:13:19 CEST
* [o] Jak2
  * [X] finish the implementation of the bond type in bond refiners
* [o] Key Discovery
  * [X] print CIKM 2011 paper
  * [ ] read CIKM 2011 paper
* [X] General
  * [X] write dr
  * [X] print train tickets

2012-08-24 Fri 09:20:05 CEST
* [X] Jak2
  * [X] run experiments
    * [X] without atom typing --> cannot grow a DT with that!
    * [X] with atom typing
    * [X] report results
  * [X] send email to everyone
* [X] General
  * [X] write dr
  * [X] progress on RoR tutorial
  * [X] think about structure of RoR models
  * [X] train tickets Leuven-Caen

2012-08-23 Thu 09:51:07 CEST
* [X] Jak2
  * [X] find papers using muta dataset
  * [X] write them into a list and make an archive
  * [X] find DT-based approaches using muta datasets
  * [X] run DT learner with classification threshold
* [X] Key Discovery
  * [X] index datasets with Jena
  * [X] run algorithm
* [X] General
  * [X] write dr

2012-08-22 Wed 10:00:04 CEST
* [X] Jak2
  * [X] add internal bond refiner
  * [X] check conformation distance list option: why does it lead to a segmentation fault?
  * [X] test interaction labels with 2-step refiner and without bond refiners (running on pinac28)
* [X] Key discovery
  * [X] Add references to planning
  * [X] Send planning to Madalina and me today
  * [X] download François's code
* [X] General
  * [X] write dr
  * [X] change date format for todo wiki

Tue Aug 21 08:10:00 CEST 2012
* [X] Test Mutag dataset
  * [X] Convert dataset from Jan
  * [X] Run DT learner on it
* [X] Prepare Skype meeting
* [X] Skype Meeting
* [X] Send an email to Jan for meeting tomorrow with Bertrand and Bruno
* [X] vim date shortcut ,D : propose other formats

Mon Aug 20 08:57:26 CEST 2012
* [X] Add bond edges
  * [X] in the form of new edge label
    * [X] copy ExternalInDistance
  * [X] implement the pattern matcher operator
    * [X] are two atoms bonded in a conformation?
      * [X] for each conformation, 
* [X] Answer Bertrand's email

Fri Aug 17 10:08:44 CEST 2012
* [X] JAK2
  * [X] Test the conformation scores
    * [X] test current implementation
      * [X] test with it
      * [X] test without it
    * [X] test a selection of best conformations only (in preprocessing)?
  * [X] Test protein/ligand interaction labels
    * [X] without labels
    * [X] with 2 types of interaction type labels
  * [X] Meeting with Jan
    * [X] xls file with RMSE and ROC score for different experiments
      * [X] with / without conformation score tests
      * [X] top1 / top5 / all conformations
      * [X] with / without interaction labels
        * [X] without
        * [X] with interaction type labels
    * [X] file with conformation scores
      * [X] add header
  * [X] Go and look for administrative documents in mailbox

Thu Aug 16 09:02:45 CEST 2012
* [X] Write planning (Madalina)
* [X] Send file with ECAI newsletter (Madalina)
* [X] Meeting Jan
* [X] Call Les Doges
* [X] Mail to Groot Begijnhof
* [X] send Jan the lipophilicity rules and some papers about decoys
* [o] JAK2
  * [X] use conformation score as test
    * [X] put score in database (preprocess and db reading)
    * [X] create new label?
    * [X] test it
  * [X] write table with scores and conformations and IC50
    * [X] in preprocessing
  * [ ] test different label sets (protein/ligand interactions)
    * [ ] how to use old labels?
* [X] write dr
* [X] send email Bruno
  * [X] ask for skype meeting
  * [X] departure date

Tue Aug 14 10:24:01 CEST 2012
* [ ] Tests with decoys and new interactions
* [ ] Mail all the group
* [X] answer Begijnhof

Thu Jul 19 09:40:22 CEST 2012
* [X] JAK2
  * [X] Copy data
  * [X] Have the last version with "old" anchor labels
  * [X] run one DT
  * [X] send to Ronan
  * [X] make a point with Ronan about the datasets available

Wed Jul 18 10:50:24 CEST 2012
* [ ] JAK2
  * [ ] Experiment 2-step refiner
    * [ ] with/without anchor (DF50, MaxDist6, samp5, K4)

Tue Jul 17 09:04:15 CEST 2012
* [X] Meeting Ronan
  * [X] Remember the work we did?
  * [X] New introduced labels
    * [X] element name, hybridization, aromaticity, charge
    * [X] HBD, HBA, lipophilicity
  * [X] Results
    * [X] Tried on other datasets
      * [X] while other approaches do not propose descriptive results, we cannot be better
      * [X] Random forest : instead of having only one DT for one dataset, we can have describing the same data (randomization)
      * [X] Protein data
  * [X] Protein Data
    * [X] Summary
      * [X] We can use the approach to exploit both the ligand data (conformations) and the protein data in the same process.
      * [X] Seem to be quite new : interesting for us, as we can show that it is possible to integrate a lot of different information.
    * [X] Questions
      * [X] Given the fact that the protein site has been considered as static, is it a good idea to use these protein data?
      * [X] Is it possible to have better data?
      * [X] Other datasets?
      * [X] Relevant distance thresholds?
      * [X] Filtering some protein atoms?
  * [X] Visualisation tool
    * [X] Summary
    * [X] Questions
  * [X] What next?

Mon Jul 16 10:33:13 CEST 2012
* [o] Experiments
  * [X] with K=4
  * [X] with Mohr data and random forests
  * [ ] with varying the sample rate
* [ ] test the correcteness of the matching operator
  * [ ] for each root anchor test and ligand 
    * [ ] calculate the distance between the anchor atom and each ligand atom, and increment the correct counter

Fri Jul 13 08:42:58 CEST 2012
* [.] test matching operator
  * [o] for each root anchor test and ligand 
    * [ ] calculate the distance between the anchor atom and each ligand atom
    * [X] compute the average values for satisfying/not satisfying ligands
      * [X] in the learning set
      * [X] in the testing set
* [X] test with simple DT and count the signs differences
* [ ] run tests with the random forest and Mohr's datasets
* [X] Jan:
  * [X] couples of results (average values)
  * [X] results with ROC
* [o] mail to Bertrand etc.
  * [ ] results
  * [X] administration

Thu Jul 12 10:16:51 CEST 2012
* [X] Check the average values on training and testing datasets for tests at the root of trees
* [o] Compare with Mohr et al. and random forests
  * [X] print the MSE and r_pred measures

Tue Jul 10 08:56:04 CEST 2012
* [o] jak2
  * [X] classification and score computation
* [ ] GRE and Keys
  * [ ] Read one paper (the Gordian approach)
* [ ] Mails
  * [ ] Anne

Mon Jul  9 09:00:30 CEST 2012
* [X] jak2
  * [X] code with decision forest
    * [X] implement voting decision
    * [X] when learner builds the DT, it also gives the example DB
* [X] Mails
  * [X] Ronan

Tue Jul  3 09:46:00 CEST 2012
* [X] Skype meeting
* [X] Get everything working
* [X] Tests
* [X] GRE & Keys
  * [X] informally note the problems
  * [X] book a room?

Mon Jul  2 09:40:58 CEST 2012
* [X] add additional vertex labels (without working)

Fri Jun 29 10:17:50 CEST 2012
* [X] Mail to Jan
  * [X] skype meetings
  * [X] protein data
  * [X] Leander and I meeting
  * [X] vacation
* [X] Mail to Leander
* [o] Code
  * [X] check whether data are correctly read and stored in the jak2 db

Thu Jun 28 11:20:21 CEST 2012
Wed Jun 27 14:35:00 CEST 2012
Tue Jun 26 10:00:08 CEST 2012
* [X] Finish reading and filtering the protein data
  * [X] coded
  * [X] problem with filter
* [o] Read the protein data in the jak2 code
  * [X] coded
  * [o] tested
    * [X] copy new jak2 data
    * [ ] check whether are correctly read and stored in the db
* [ ] Start writing an additional possible refinement
  * [ ] of the form of a vertex label

Fri Jun 22 10:09:35 CEST 2012
* [X] read closed graph mining
* [X] geometric pattern mining
  * [X] list possible difficulties
* [.] Meeting with Jan
  * [X] Task1
    * [X] Ask Ronan, Leander, Thomas whether the information about the protein side can be used or not. How to collect the data?
  * [.] Task2
    * [o] As soon as the protein-side data are available
      * [X] Preprocessing and filter the protein atoms to find anchor atoms:
        * [X] keep only atoms that are at distance < 6-7 angstroms to a ligand atom in at least 3% or 5% of the instances --> those are anchor points
      * [ ] Expand the pattern language in order to take into account the protein binding site. 2 types of new tests in the refinement operator:
        * [ ] Given an anchor point, is there an atom that is in the neighborhood of this anchor atom?
        * [ ] Given a pattern, is there an anchor point that can be "connected" to one of the vertices of the pattern?
  * [ ] Task3
    * [ ] Reduce the memory consumption of the patterns
      * [ ] use minimum-cost types to reduce unnecessary memory consumption
      * [ ] remove the pattern-matching preprocessing features after the pattern matching
  * [ ] Task4: for the closed pattern mining problem
    * [ ] Compute hash-values out of the pattern extension: a vector of bits that represent for each conformation (or example) whether or not it covers the pattern
      * [ ] 2 patterns having the same hash-value have to be further studied
      * If two patterns have the same vector, then we should test further
    * [ ] Also use this principle for the DT learner
    * [ ] Ask Mostafa about "Hashing of root images"

Thu Jun 21 09:59:34 CEST 2012
* [X] Test new heuristic
  * [X] Run the tests
  * [X] Compare the plots RSE vs predicted/real value
* [X] Implement new heuristics
  * [X] give more weight to high values

Mon Jun 18 08:01:19 CEST 2012
* [X] Mail
  * [X] document with plotting the IC50 value against RMSE
  * [X] about new paper and data and approach
* [X] paper to read
  * [X] read paper
  * [X] data to find
  * [X] source code to find
  * [X] preprocessing scripts to find

Fri Jun 15 09:44:22 CEST 2012
* [X] Check availability of Mohr et al predictions
* [X] Send mail to group
* [X] Plot data (RMSE vs IC50)
  * [X] Create the input file for Gnuplot
* [X] Run XP:
  * [X] Everything with 
    * [X] filter hbd_hba_polar
    * [X] 2-step refiner
    * [X] Minimum leaf size
      * [X] 15
      * [X] 25
      * [X] 30

Wed Jun 13 09:02:55 CEST 2012
* [X] Remove polarity label to remove reduce the pattern nb
* [X] in Cross-Validation, change the computation of the average values when K=2
* [X] check the computation of
  * [X] SD measure
  * [X] MSE/RMSE
  * [X] ROC
* [X] In CrossValidation log, add the YES/NO refiners
* [X] New experiments without DHFR
  * [X] Labels
  * [X] Filters
* [X] Add lipophilicity
* [X] Also make experiments with 1-step refiner

Tue Jun 12 11:12:09 CEST 2012
* [X] JAK2
  * [X] is the generation of patterns okay? (check whether the same label can be used in different vertices)
  * [X] run tests for evaluating the different labels

Mon Jun 11 08:57:29 CEST 2012
* [X] JAK2
  * [X] test if everything is working
    * [X] Try on a small dataset
    * [X] in preprocessing, write the folds.data file
    * [X] test the xp scripts
  * [X] run new tests with new parameters and new information
* [X] Mail Pascal Sandra
* [X] Mail parents

Sun Jun 10 11:05:50 CEST 2012
* [.] PVS: Git project reorganisation
  * [ ] Move all projects
  * [X] Update .git/config on local copies
    * [X] vim_config
    * [X] ssh_config
  * [ ] create new projects
    * [ ] General config (scripts and files)
    * [ ] Key discovery
    * [ ] GRE
    * [ ] Inhibitor pattern mining
* [.] JAK2
  * [X] create labels
  * [ ] add lipophilic information
  * [ ] add corresponding label

Fri Jun  8 09:06:26 CEST 2012
* [X] ERC report
  * [X] write information
  * [X] send to Samrat
* [X] JAK2
  * [X] Filtering atoms
    * [X] implements different filters
    * [X] write the datasets according to it
  * [X] Convert the new datasets and the jak2 dataset
  * [X] In jak2 code, correctly read the data
* [.] Various
  * [O] Download interesting courses
    * [X] Automata
    * [X] Computer vision
    * [X] Introduction to logic
  * [ ] Notes on the key discovery problem
    * [ ] look at the 2006 paper, Gordian approach
    * [ ] look at the VLDB 2010 paper
    * [ ] notes on the related work
  * [ ] Git projects on PVS
    * [ ] a project for KeyDiscovery
    * [ ] a project for Pattern-based regression

Thu Jun  7 10:11:40 CEST 2012
* [X] JAK2
  * [X] OpenBabel integration
    * [X] For new datasets
      * [X] read input data files
        * [X] atoms and bonds information
        * [X] molecules information
      * [X] write in jak2 files
        * [X] target_values
        * [X] conformations

Wed Jun  6 10:16:56 CEST 2012
* [X] JAK2
  * [X] Check results from various datasets
  * [X] re-run tests on datasets
    * [X] update scripts
    * [X] run them
  * [X] Start OpenBabel integration
    * [X] compile and install openbabel
    * [X] create a project including OpenBabel
* [o] Various
  * [X] Notes on the key mining problem

Tue Jun  5 10:42:22 CEST 2012
* [X] JAK2
  * [X] update the script parameters.sh
  * [X] update jak2 code to consider new settings
    * [X] use log
    * [X] cross-validation fold number
    * [X] re-add discretization parameters
  * [X] update the script run.sh
  * [X] write a script to run some tests for evaluating the impact of parameters
* [X] Skype Bruno - Bertrand
* [X] Mails
  * [X] to Pascal
  * [X] to Yoann

Mon Jun  4 09:54:03 CEST 2012
* [X] JAK2
  * [X] memory problem
    * [X] manually refining a pattern
    * [X] create new example
    * [X] refine patterns until a certain number of patterns
  * [X] preprocessing data
  * [X] filtering atoms when reading the data
* [o] Other
  * [X] mail to Bertrand and Bruno

dimanche 3 juin 2012, 15:25:48 (UTC+0200)
* [X] JAK2
  * [X] in Cross Validation, create an empty pattern

Wed May 30 09:03:48 CEST 2012
* [X] UM3
  * [X] print:
    * [X] train tickets
    * [X] dossier
    * [X] information on MISASHS
    * [X] slides
    * [X] information on various projects
* [.] JAK2
  * [X] print node information (description)
  * [ ] Scripts to run on JAK2 data (2-step refinement)
  * [ ] Scripts to run on other datasets (2-step refinement)
    * [ ] Preprocessing all files (for regression)
    * [ ] Creating training/test datasets

mardi 29 mai 2012, 08:30:41 (UTC+0200)
* [X] UM3 presentation
  * [X] check the content
  * [X] send it
* [X] Skype Meeting
  * [X] Results
    * [X] on other datasets
      * [X] Results to analyse and compare
      * [X] Do it with the other datasets
      * [X] Comparison
        * [X] what is the measure used in the paper?
    * [X] 2-step refiner
      * [X] no statistics on the failing
      * [X] new strategy?
        * [X] not storing the patterns
        * [X] looping over the patterns before the ligands

lundi 28 mai 2012, 09:16:03 (UTC+0200)
 * [X] Audition UM3
   * [X] Discours slide par slide
 * [X] JAK2
   * [X] when does a pattern is destructed?
   * [X] pattern selection in the query engine

Sat May 26 10:28:03 CEST 2012
* [X] jak2
  * [X] preprocess 5 datasets
  * [X] run tests on explorer
    * [X] 1-step refiner
    * [X] 2-step refiner
* [X] Work on memory consumption issues
  * [X] Remarks:
   - replaced the testCollection by the DBWrapper
   - patterns and corresponding tests are generated, but they are not selected ? --> resulting DT only contain the root
   - when counting the nb of patterns generated and destructed, there is a gap : not all patterns are destructed, where they should be destructed just before creating a new one
* [X] change audition presentation

Fri May 25 14:26:33 CEST 2012
* [ ] take into account Pascal and Sandra's remarks on presentation
* [o] work on jak2 problem
  * [X] correct it
  * [X] push the code
  * [X] run tests on Ronan's data (2-step refiner) on pinacs
* [ ] mail Yoann
* [X] print train tickets
* [X] print 2 papers from Bertrand

Thu May 24 07:49:05 CEST 2012
* [X] Paper
  * [X] read paper
    - QSAR based on the similarity between 3D structures (molecule kernel, independent of the spatial prealignement of the compounds)
    - Kernel + potential SVM
    - Relying on pairs of ordered bipods (V-shaped subfragments)
    - Related work compared with the approach:
      - Hologram QSAR (HQSAR) --> divides each molecule into a set of overlapping structural fragments and uses their frequency as 2D descriptors
      - Comparative molecular field analysis (CoMFa) --> 3D QSAR, based on the idea similar steric and electrostatic fields of molecules lead to similar activity. Requires the spatial-imposition of the molecular structures, which needs expert knowledge and is time-consuming and might introduce user-bias
      - CoMSIA (comparative molecular similarity index analysis) --> altervative method to CoMFa
      - GRIND method: 3D QSAR --> only applicable for a set of compounds sharing a common scaffold
      - EVA (eigenvalue analysis)
    - molecular alignment = rigid transformation of the coordinates of a molecule
    - DATASETS
      - Fontaine (classification): 435 molecules (156 low-, 279 high-activity)
      - ACE (regression): 114 molecules
      - AChE (regression): 111 molecules
      - BZR (regression): 163 molecules
      - DHFR (regression) 397 examples
  * [X] are datasets publicly available?
  * [X] how the paper is cited?
    * [X] 9 times
* [X] Experiments
  * [X] Compare with results in paper
* [X] UM3
  * [X] send pdf document
  * [X] send email to Christian Lavergne
  * [X] send email to Nicolas (infos)
  * [X] répondre Bruno

Wed May 23 08:00:15 CEST 2012
* [O] UM3
  * [X] sélectionner publications
    * [X] lettre signée avec résumé
    * [X] mail Nicolas (nombre d'exemplaires, thèse?)
  * [.] finir présentation
    * [X] trouver et stocker exemples de présentations
    * [ ] enseignement
    * [ ] recherche
  * [X] Mail Sandra : utile de contacter Christian Lavergne?
* [X] summary of last meeting to Thomas
  * [X] mail
* [o] JAK2
  * [O] compare with the results of the genetic algo paper
    * [X] in reading the input data, change IC50 to 1 or 0 whether it is above/below the given threshold --> 0 is inactive, 1 is active
    * [X] remove log(IC50)
    * [ ] note results in classification
    * [X] new Test with memory problem fixed
      * [X] on Ronan data --> pinacs
      * [X] on bzr_3d data: with 1-step refiner --> test in a classification context
  * [ ] look at http://pubs.acs.org/doi/abs/10.1021/ci800144y, do they have a publicly available dataset? 
  * [X] memory problem
    * [X] replace shared_ptr<pattern> by weak_ptr<pattern> --> not possible without more work
    * [X] found and fixed

Tue May 22 09:22:11 CEST 2012
* [X] JAK2
  * [X] test with/without keeping alternative patterns
    * [X] Result: no difference at all between keeping and not keeping intermediate patterns
  * [X] explicitely clear the test collection of a node when it has been finished (in mips)
    * [X] clear the container (DBWrapper == TestCollection) when all tests have been checked
    * [X] print memory usage before and after --> no effect at all on the destruction of the patterns
    * [X] find a way to clear the patterns when the test collection is cleared
      * [X] Problem: in a test, the pattern is associated with a boost::shared_ptr --> maybe even when the test collection is cleared, the patterns are pointed by more than one object (i.e., by at least another object than the test itself) --> yes, there are several references to the same pattern (2 at the beginning, 3 after).
      * [X] Is there a case where there is only 1 reference to a pattern? --> apparently not. So, why patterns are sometimes destructed?
      * [X] in removeUsedTestGenerator.hpp it seems that the test collection of a parent node is copied and passed to children? --> not used for jak2
      * [X] print the number of references to a pattern
      * [X] in embeddingTest (jak2), create a delete function
      * [X] is the number of references to a pattern the same for 1 and 2-step refiners?

Mon May 21 09:05:03 CEST 2012
* [X] UM3
  * [X] mail Nicolas 
  * [X] préparer dossier pdf
* [X] Jak2
  * [X] Code
    * [X] Test the memory consumption with 1-step refiner
    * [X] Create a destructor for the patterns and print something when used
    * [X] Speak with Thomas
  * [X] Answer Bruno's email

Mon May 14 10:41:56 CEST 2012
* [X] Mail Pascal et Sandra
* [X] Bertrand and Ronan
  * [X] Read Muggleton 2008 paper (in particular, the data description)
  * [X] Discuss it with Thomas
* [o] UM3
  * [X] Mail to Nicolas
  * [ ] Create PDF document
* [ ] Mail Jan
  * [ ] results of experiments (how long for learning the first DT?)
  * [ ] Audition in Montpellier
* [ ] Mail Yoann
* [ ] JAK2 project
  * [ ] convert the datasets of JCIM2003 (Spline-fitting with a genetic algorithm, Sutherland et al...)
  * [ ] look at what has been done by Muggleton
  * [ ] integrate angles in constraints?
* [X] Text prediction: 
  * [X] execute FPGrowth

Wed May  9 08:52:18 CEST 2012
* [X] select best patterns in trees
* [X] push new code on git
* [X] put experiments on svn
* [ ] look at other datasets with 3D information
* [X] look at the impact of the equalWidth discretizer
* [X] add negative examples to Muggleton's data (by removing atoms of the pharmacophore) --> cancelled, we will wait for another dataset
* [ ] change format of experiments summary --> in csv
* [X] run with no grounded at labels and 2-levels deep lookahead

Tue May  8 08:56:23 CEST 2012
* [X] give all the embeddings of one given pattern in one given ligand
  * [X] in subgraph isomorphism, store all the possible embeddings
  * [X] in ConjunctiveSubgraphIsomorphism, count and write (in an outstream) all the possible embeddings
* [X] run DT learner on Muggleton's data
  * [X] equivalent of IC50 value?

Mon May  7 10:27:15 CEST 2012
* [X] Convert Muggleton data (eclipse)
  * [X] read atoms and bonds
    * [X] add charge information --> everything to 0
    * [X] add aromaticity information
  * [X] write cvs files
* [X] Download courses
* [X] Experiments: run with equalWidth interval distances, and smaller interval size, and overlapping intervals (1, 2, 3, 4)
* [X] Mail Bruno and Bertrand
* [X] install mail kuleuven on phone
* [ ] Interesting paths in decision trees
* [X] mutt configuration
  * [X] added gmail address
* [X] separate jak2 code for experiments and coding
* [X] move scripts and results in the new folder for experiments

Fri May  4 10:46:51 CEST 2012
* [X] phone
  * [X] charge
  * [X] files
* [X] Answer mail Sandra, Pascal 
* [X] Mail UM3
* [X] Printing 2 chapters from Graph Theory book
* [X] Start transforming Muggleton's dataset

jeudi 3 mai 2012, 14:31:16 (UTC+0200)
* [X] fix problem with discretizer type preprocessor
* [X] added parameters for overlapping interval number in settings
* [X] fix problem with ROC computing

mercredi 2 mai 2012, 09:42:11 (UTC+0200)
* [X] Mail réponse Sandra
* [X] Put first results on svn repository
* [X] Discretization based on equal-width intervals
  * [X] allow overlapping intervals for all distance discretizers
    * [X] method discretize returns a list of intervals
    * [X] implement discretize for
      * [X] Geometric discretizer (only for 1 overlapping interval)
      * [X] EqualWidth discretizer
    * [X] Check if the new discretization works well with preprocessings (distance lists, etc.)
* [X] Look for other measures for test selection in the DT learner

mercredi 25 avril 2012, 15:01:54 (UTC+0200)
* [O] Implement and count ROC-like value
  * [X] in 1 DT, count all couples of ligands
  * [X] in 1 DT, count only couples of consecutive ligands
  * [ ] extend to the cross-validation (average)
  * [X] extend to the cross-validation (taking into account all possible ligands)
  * [X] integrate into the scripts
    * [X] java post-processer read and write
* [X] Logs all experiments on trantor -> add the date
* [X] Print the settings in the DT log
* [X] Print the predictions in the DT log
* [X] In HTML summary of experiments
  * [X] add the date
  * [X] add links to the log files
* [X] Add to experiment scripts:
  * [X] an argument to choose the type of distance discretization
  * [X] an usage section
* [ ] Convert Muggleton data into our format + try DT learner
* [X] Distance discretization based on equal-width intervals
  * [X] allow an overlapping of intervals
* [ ] Count embeddings of a pattern in a ligand (in each conformation) --> to be used in Leander visualization tool
* [ ] Documentation Doxygen

* [ ] AKDM3 
  * [ ] mail to organizers
  * [ ] mail to Pascal, Sandra
* [o] JAK2  
  * [ ] run tests
  * [X] add the variance reduction and information gain (avg and standard deviation) in:
    * [X] log
    * [X] scripts
    * [X] java post-processing
  * [X] log global results in a HTML page
* [X] install jak2 code on explorer

jeudi 19 avril 2012, 09:35:36 (UTC+0200)
* [X] Mail Hassan, Yoann

mercredi 18 avril 2012, 10:06:54 (UTC+0200)
* [X] List of things to read
* [ ] Message for Ronan?
* [ ] Data on Caen computer
* [o] Mails:
  * [X] Pascal, Sandra 
  * [ ] AKDM3

vendredi 13 avril 2012, 09:21:17 (UTC+0200)
* [O] Code
  * [X] external class cross-validation
    * [X] log results
		* [X] log the folds
	* [X] in settings, add the interesting leaves IC50 threshold
  * [ ] have results for interesting leaves only (IC50 < 100)
  * [X] external class find embeddings
  * [X] external class leave sorter
  * [X] Experimental protocol
    * [X] script to:
      * [X] change the settings in settings.hpp 
			* [X] run the code (compilation+running)

mardi 10 avril 2012, 08:10:51 (UTC+0200)
* [ ] contract --> send to Annemie Dumoulin
* [X] email with address and bank account
* [ ] RMSE global in the DT and for interesting leaves only (IC50 < 100)
  * [ ] multiply by the number of examples
* [ ] Benchmarks
  * [ ] find datasets
  * [ ] ask Leander for datasets and methods to compare with
* [X] mail Nicolas

lundi 9 avril 2012, 10:48:46 (UTC+0200)
* [X] get clang_complete plugin working
* [X] dr
* [X] pattern matcher code
  * [X] use connected_components from boost
  * [X] optimize the methods using the nth vertex
    * [X] use the access to vertices and edges g[edgeOrVertex]
      * [X] in vertexList option

Fri Apr  6 11:00:02 CEST 2012
* [ ] Pattern matching with unconnected graphs (conjunctions of graphs)
* [X] Log embeddings
  * [X] as a post-processing
  * [X] INPUT: pattern / OUTPUT : list of embeddings
  * [X] in OUTPUT log, print a pattern with the code of the node it is associated with (optional)
* [X] Pattern OUTPUT
  * [X] as a list of logical conjunctions (e.g., GroundedAt(X,4,2.3,1),InDistance(X,Y,6.7),...)
* [o] PATTERN INPUT/OUPUT
  * [X] GraphViz output
* [.] TESTS
  * [o] Allow a 10-fold cross-validation
    * [X] automatic random segmentation of the dataset
    * [ ] running and storing the correct information
  * [ ] As INPUT : the pattern language wanted for the execution
  * [ ] As OUTPUT :
    * [ ] average time (and +/- standard deviation?)
    * [ ] RMSE (rooted mean squared error) --> in multi-instances settings, use the conformation with the smallest IC50 value
    * [ ] average tree size +/- standard deviation
    * [ ] log everything
      * [ ] the datasets resulting from the cross-validation?

Wed Apr  4 09:41:01 CEST 2012
* [ ] Document about list of experiments
  * [ ] RMSE instead of MSE
  * [ ] average time on cross-validation and size of the tree

Tue Apr  3 09:18:28 CEST 2012
* [X] Meeting with Jan, Thomas, Bertrand
* [X] Updating the code	on local machine
* [X] installing QT-creator on local machine
* [X] Administration
  * [X] Groot Begijnhof
    * [X] contract Groot Begijnhof
    * [X] inventory
    * [X] contacting bank for the bank transfer

lundi 26 mars 2012, 08:49:12 (UTC+0200)
* [ ] Dossier candidature
  * [ ] ajouter école de printemps?
  * [ ] 
* [X] Mail Thomas accommodation
* [ ] Code jak2
* [X] Mail Mathieu Roche

dimanche 25 mars 2012, 14:03:27 (UTC+0200)
* [o] Dossier UM3
  * [ ] Dossier enseignement
    * [ ] Détail Fouille de Texte : ajouter lien vers supports de cours
    * [ ] Ajouter Béziers et insister sur 'je connais ...'
    * [ ] Parler des différentes formations
  * [X] Dossier recherche
    * [X] Finir texte
  * [o] Divers
    * [X] Ajouter jury de thèse en page 2
    * [X] Ajouter Leuven en page 1 (CV)
    * [ ] Ajouter intervenants

vendredi 23 mars 2012, 08:09:23 (UTC+0100)
* [o] Dossier candidature MCF UM3
  * [X] Relire projet enseignement
  * [X] Envoyer à Sandra
  * [ ] Ecrire projet de recherche
  * [ ] Envoyer à Pascal
* [X] Corrections Fouille de texte
* [ ] Papier AKDM
* [X] Rappeler Bruno et Bertrand (lettre de recommandations)

mercredi 21 mars 2012, 08:35:04 (UTC+0100)
* [X] Final version of AKDM paper
* [X] mail Maguelonne
* [X] mail Mathieu
* [X] mail Marc
* [X] mail Enseignements UCBN
  * [X] envoyer CV
* [X] List of experiments on Jak2
  * [X] Ronan
    * [X] One and Two-levels deep refinement strategy
    * [X] Distance upper limit --> TO IMPLEMENT
    * [X] Finer discretization on distances
    * [X] With/without a strategy for reducing IC50 value in Yes branches --> TO IMPLEMENT
  * [X] Paper
    * [X] IC50 value, log(IC50), rank
    * [X] Prediction accuracy (with training and test datasets)
    * [X] Different discretization methods:
      * [X] Distances
        * [X] based on a geometric series interval bounds (current)
        * [X] based on equal-frequency intervals
        * [X] based on equal-length intervals
      * [X] Coordinates
        * [X] different values on boxes size
        * [X] space-segmentation methods
    * [X] Pre/Post-pruning parameters in the decision tree?
* [X] Jak2 code
  * [X] distance upper limit
  * [X] decreasing IC50 value
* [X] Summary document on Jak2 project for Bruno
* [X] Dossier MCF : Enseignement
* [X] Corrections AKDM Sandra

dimanche 18 mars 2012, 09:55:10 (UTC+0100)
* [X] Mail KUL (admin)
* [X] Dossier UM3
  * [X] CV
  * [X] Préparer mail
  * [X] Envoyer mail
* [X] Code
  * [X] supprimer vertexList

lundi 12 mars 2012, 08:45:50 (UTC+0100)
* [X] Mails:
  * [X] Accommodation Leuven
  * [X] Bertrand : plan présentation
  * [X] Thomas : 1 level-deep
  * [X] Sandra : recommandation letter
  * [X] Admin : GREYC web site

dimanche 11 mars 2012, 15:14:17 (UTC+0100)
* [X] mail Sandra
  * [X] corrections lettre + remarque
  * [X] n° de poste
* [X] Cours 3 Fouille de texte

mercredi 7 mars 2012, 17:21:53 (UTC+0100)
* [X] Presentation preparation for IATE on 2012-03-09
  * [X] S6: better link between association rules and FCA?
  * [X] S6: "the attributes are directly viewed as itemsets" --> "the attributes are directly viewed as items"
  * [X] S7: not "translating the FCA problem into a frequent itemset mining problem"?
  * [X] S7: automatic ontology elicitation?
  * [X] S30: put an image from the last presentation?
  * [X] S32: first line : ????! :)
  * [X] At the beginning, make a slide on the itemset mining problem (quite simple) in order to better see the translation into FCA
    * [X] Quick presentation of the problem --> example(s)
    * [X] Show the link with knowledge acquisition
  * [X] Implications in a concept lattice
    * [X] have a good understanding of it: e.g., what happens with dog?
    * [X] introduce the fact that association rules can also provide this type of rules, but with a less "crisp" notion of implication
  * [X] How to introduce sequential pattern mining?
  * [X] How to introduce contextual pattern mining?
    * [X] think about the exploitation of contextual pattern mining in itemsets : what can it give that classical pattern mining can't? Domain knowledge can be exploited in order to build the context hiearchy
  * [X] When showing items in sequences that are related to originally numerical attributes, speak about fuzzy logic and pattern mining techniques?
  * [X] Show one context hierarchy in the slides?
  * [X] Globally highlight the fact that extracting patterns is usually very good when dealing with experts : they are interpretable (like the original idea of fuzzy logic?.. :) )
  * [X] Find a concrete example of the fact that contextual information can be seen at a different level than attributes-values
  * [X] when dealing with sequential patterns, mention that it is about dealing with structured data (sequences, graphs, etc.), which is a more complex domain than itemsets, but that can model numerous real-world domains (e.g., time-related domains, bioinformatics, texts, chemistry, etc.). Mention the fact that I already have worked on numbers of them.
  * [X] Two directions: from pattern mining to knowledge and "from knowledge (ontologies, taxonomies) to patterns to applications"
    * [X] Direction 1
      * [X] fastly provide an ontology based on concepts taht can be observed within the data (then validation)
    * [X] Direction 2
      * [X] being able to extract patterns at a given level of a taxonomy or directly exploiting the different levels of a taxonomy (without a specified one)
      * [X] use the obtained knowledge to build the contexts that will be used in order to obtain precise results

mardi 6 mars 2012, 15:16:38 (UTC+0100)
* [X] Presentation in chemoinformatic group
* [X] Document for GREYC administration (mission document)
* [ ] Thomas
  * [ ] code running?
  * [ ] meeting with Ronan? Coming to Caen?
* [o] Leuven
  * [ ] registration on web site
  * [X] username and password
  * [X] documents
    * [X] CV
    * [X] passport
    * [X] diploma
* [X] Mail Hugo
* [ ] Teaching
  * [ ] Lectures
    * [ ] 
  * [ ] Practical work
* [X] GREYC web site

lundi 5 mars 2012, 15:52:40 (UTC+0100)

  * [X] Arbre de décision
    * [X] Exemple + explications
    * [X] Avantage : problème de seuil
  * [X] Exemple motif --> animation
    * [X] créer images pour BD
    * [X] créer images pour motif
  * [ ] Mettre en forme
  * [X] Remplacer la section "résultats expérimentaux" par autre chose ?
  * [X] To add
    * [X] logos on slide 1
    * [X] slide numbers
    * [X] unité pour l'IC50 (nM)

vendredi 2 mars 2012, 09:34:59 (UTC+0100)
* [X] mail réponse Bertrand
* [X] mail Thomas
* [X] any idea for pattern matcher time?
* [X] advances on presentation

jeudi 1 mars 2012, 15:36:06 (UTC+0100)

* [X] réunion Ronan CERMN
* [X] mail Bertrand
* [X] Code
  * [X] problem with distances
  * [X] check and insert rules for hybridization
  * [X] logarithmic IC50 value
* [ ] Having jak2 and mips code running on laptop

mercredi 29 février 2012, 14:52:19 (UTC+0100)

* [X] Code
* [X] Questions Ronan
* [X] Mail Thomas
* [X] Mail Bruno (déplacement Paris)


vendredi 24 février 2012, 08:43:10 (UTC+0100)
* [X] cours
	- [X] imprimer documents utiles
	- [X] mail Gaël Dias, Frédérick
	- [X] plan 1er cours + demander énoncer TP Nicolas


= GDR I3 presentation: outline =
* I. Contexte chémogénomique
	- Affinité protéique
	- Voies de dégradation
	- Application : inhiber JAK, découvrir des ligands
	- Montrer une image qui montre l'affinité d'une protéine et d'un ligand (conformation)
* II. Information recherchée (Motifs recherchés)
	- Type de motifs
	- Arbre de décision avec motifs
	- Aspect multi-instances (différentes conformations --> 1 slide : un ligand satisfait un motif si une des conformations satisfait le motif)
* III. Méthode d'extraction
	- Ne pas focaliser dessus pour le 6 mars
* IV. Résultats expérimentaux
	- Temps en fonction de certains paramètres
	- Avec ou sans utilisation de la stratégie de refinement sur le nouveau sommet généré
	- Nombre de motifs générés
* V. Conclusion et perspectives
	- Outil de visualisation des motifs dans les conformations pour les experts, outil de paramétrage
	
mardi 28 février 2012, 10:36:49 (UTC+0100)
* [X] Finir Cours + TP
  * [X] cours
    * [X] logos
    * [X] corrections
    * [X] other exercices?
    * [X] avantages des motifs : interprétabilité, tiennent pleinement compte de l'aspect séquentiel des données textuelles, peuvent être considérés comme de nouveaux descripteurs d'un texte (classification, ...)
    * [X] notation
  * [X] TP
    * [X] pas à pas :
      * [X] compiler le code
      * [X] extraire sans filtrage et lemmatisation
      * [X] extraire sans filtrage et avec les étiquettes grammaticales
      * [X] traduire les motifs
      * [X] extraire avec filtrage grammatical
      * [X] extraire avec lemmatisation
* [O] JAK2 Data
  * [X] create list of distances (Java)
  * [X] create list of coordinates (on x, y , z) (Java)
  * [X] create histograms (Weka, R, Gnuplot, ?)
  * [.] write short document
    * [ ] different parameters for the discretization
    * [X] histograms and data
      * [X] minimum and maximum values, mean value, etc.
      * [X] density histograms
    * [ ] how to select the right parameters?
* [X] email to Ronan --> ok for thursday
  * [X] first patterns (decision tree)
  * [X] book + usb key
  * [X] data to tag
  * [X] H-bond donor / acceptor?

* [o] Maison
  * [X] poubelle salon
  * [X] poubelle salle de bain
  * [ ] nettoyer table
  * [ ] nettoyer douche
  * [ ] nettoyer sol?
