====== General table of content ======
[[General TODO]]
[[Mails]]
[[To learn and read]]
[[JAK2]]
[[General_Ideas]]
[[PaperReading]]
[[KULeuven information]]
[[TextPrediction]]
[[Paper Writing]]
[[Web development]]
[[Various]]

== Daily TODO ==

Fri Jul 20 09:18:35 CEST 2012
* [ ] JAK2
  * [ ] new preprocessing
  * [ ] new DT and embeddings
* [ ] Various
  * [ ] email to Anne
Thu Jul 19 09:40:22 CEST 2012
* [X] JAK2
  * [X] Copy data
  * [X] Have the last version with "old" anchor labels
  * [X] run one DT
  * [X] send to Ronan
  * [X] make a point with Ronan about the datasets available

Wed Jul 18 10:50:24 CEST 2012
* [ ] JAK2
  * [ ] Experiment 2-step refiner
    * [ ] with/without anchor (DF50, MaxDist6, samp5, K4)

Tue Jul 17 09:04:15 CEST 2012
* [X] Meeting Ronan
  * [X] Remember the work we did?
  * [X] New introduced labels
    * [X] element name, hybridization, aromaticity, charge
    * [X] HBD, HBA, lipophilicity
  * [X] Results
    * [X] Tried on other datasets
      * [X] while other approaches do not propose descriptive results, we cannot be better
      * [X] Random forest : instead of having only one DT for one dataset, we can have describing the same data (randomization)
      * [X] Protein data
  * [X] Protein Data
    * [X] Summary
      * [X] We can use the approach to exploit both the ligand data (conformations) and the protein data in the same process.
      * [X] Seem to be quite new : interesting for us, as we can show that it is possible to integrate a lot of different information.
    * [X] Questions
      * [X] Given the fact that the protein site has been considered as static, is it a good idea to use these protein data?
      * [X] Is it possible to have better data?
      * [X] Other datasets?
      * [X] Relevant distance thresholds?
      * [X] Filtering some protein atoms?
  * [X] Visualisation tool
    * [X] Summary
    * [X] Questions
  * [X] What next?

Mon Jul 16 10:33:13 CEST 2012
* [o] Experiments
  * [X] with K=4
  * [X] with Mohr data and random forests
  * [ ] with varying the sample rate
* [ ] test the correcteness of the matching operator
  * [ ] for each root anchor test and ligand 
    * [ ] calculate the distance between the anchor atom and each ligand atom, and increment the correct counter

Fri Jul 13 08:42:58 CEST 2012
* [.] test matching operator
  * [o] for each root anchor test and ligand 
    * [ ] calculate the distance between the anchor atom and each ligand atom
    * [X] compute the average values for satisfying/not satisfying ligands
      * [X] in the learning set
      * [X] in the testing set
* [X] test with simple DT and count the signs differences
* [ ] run tests with the random forest and Mohr's datasets
* [X] Jan:
  * [X] couples of results (average values)
  * [X] results with ROC
* [o] mail to Bertrand etc.
  * [ ] results
  * [X] administration

Thu Jul 12 10:16:51 CEST 2012
* [X] Check the average values on training and testing datasets for tests at the root of trees
* [o] Compare with Mohr et al. and random forests
  * [X] print the MSE and r_pred measures

Tue Jul 10 08:56:04 CEST 2012
* [o] jak2
  * [X] classification and score computation
* [ ] GRE and Keys
  * [ ] Read one paper (the Gordian approach)
* [ ] Mails
  * [ ] Anne

Mon Jul  9 09:00:30 CEST 2012
* [X] jak2
  * [X] code with decision forest
    * [X] implement voting decision
    * [X] when learner builds the DT, it also gives the example DB
* [X] Mails
  * [X] Ronan

Tue Jul  3 09:46:00 CEST 2012
* [X] Skype meeting
* [X] Get everything working
* [X] Tests
* [X] GRE & Keys
  * [X] informally note the problems
  * [X] book a room?

Mon Jul  2 09:40:58 CEST 2012
* [X] add additional vertex labels (without working)

Fri Jun 29 10:17:50 CEST 2012
* [X] Mail to Jan
  * [X] skype meetings
  * [X] protein data
  * [X] Leander and I meeting
  * [X] vacation
* [X] Mail to Leander
* [o] Code
  * [X] check whether data are correctly read and stored in the jak2 db

Thu Jun 28 11:20:21 CEST 2012
Wed Jun 27 14:35:00 CEST 2012
Tue Jun 26 10:00:08 CEST 2012
* [X] Finish reading and filtering the protein data
  * [X] coded
  * [X] problem with filter
* [o] Read the protein data in the jak2 code
  * [X] coded
  * [o] tested
    * [X] copy new jak2 data
    * [ ] check whether are correctly read and stored in the db
* [ ] Start writing an additional possible refinement
  * [ ] of the form of a vertex label

Fri Jun 22 10:09:35 CEST 2012
* [X] read closed graph mining
* [X] geometric pattern mining
  * [X] list possible difficulties
* [.] Meeting with Jan
  * [X] Task1
    * [X] Ask Ronan, Leander, Thomas whether the information about the protein side can be used or not. How to collect the data?
  * [.] Task2
    * [o] As soon as the protein-side data are available
      * [X] Preprocessing and filter the protein atoms to find anchor atoms:
        * [X] keep only atoms that are at distance < 6-7 angstroms to a ligand atom in at least 3% or 5% of the instances --> those are anchor points
      * [ ] Expand the pattern language in order to take into account the protein binding site. 2 types of new tests in the refinement operator:
        * [ ] Given an anchor point, is there an atom that is in the neighborhood of this anchor atom?
        * [ ] Given a pattern, is there an anchor point that can be "connected" to one of the vertices of the pattern?
  * [ ] Task3
    * [ ] Reduce the memory consumption of the patterns
      * [ ] use minimum-cost types to reduce unnecessary memory consumption
      * [ ] remove the pattern-matching preprocessing features after the pattern matching
  * [ ] Task4: for the closed pattern mining problem
    * [ ] Compute hash-values out of the pattern extension: a vector of bits that represent for each conformation (or example) whether or not it covers the pattern
      * [ ] 2 patterns having the same hash-value have to be further studied
      * If two patterns have the same vector, then we should test further
    * [ ] Also use this principle for the DT learner
    * [ ] Ask Mostafa about "Hashing of root images"

Thu Jun 21 09:59:34 CEST 2012
* [X] Test new heuristic
  * [X] Run the tests
  * [X] Compare the plots RSE vs predicted/real value
* [X] Implement new heuristics
  * [X] give more weight to high values

Mon Jun 18 08:01:19 CEST 2012
* [X] Mail
  * [X] document with plotting the IC50 value against RMSE
  * [X] about new paper and data and approach
* [X] paper to read
  * [X] read paper
  * [X] data to find
  * [X] source code to find
  * [X] preprocessing scripts to find

Fri Jun 15 09:44:22 CEST 2012
* [X] Check availability of Mohr et al predictions
* [X] Send mail to group
* [X] Plot data (RMSE vs IC50)
  * [X] Create the input file for Gnuplot
* [X] Run XP:
  * [X] Everything with 
    * [X] filter hbd_hba_polar
    * [X] 2-step refiner
    * [X] Minimum leaf size
      * [X] 15
      * [X] 25
      * [X] 30

Wed Jun 13 09:02:55 CEST 2012
* [X] Remove polarity label to remove reduce the pattern nb
* [X] in Cross-Validation, change the computation of the average values when K=2
* [X] check the computation of
  * [X] SD measure
  * [X] MSE/RMSE
  * [X] ROC
* [X] In CrossValidation log, add the YES/NO refiners
* [X] New experiments without DHFR
  * [X] Labels
  * [X] Filters
* [X] Add lipophilicity
* [X] Also make experiments with 1-step refiner

Tue Jun 12 11:12:09 CEST 2012
* [X] JAK2
  * [X] is the generation of patterns okay? (check whether the same label can be used in different vertices)
  * [X] run tests for evaluating the different labels

Mon Jun 11 08:57:29 CEST 2012
* [X] JAK2
  * [X] test if everything is working
    * [X] Try on a small dataset
    * [X] in preprocessing, write the folds.data file
    * [X] test the xp scripts
  * [X] run new tests with new parameters and new information
* [X] Mail Pascal Sandra
* [X] Mail parents

Sun Jun 10 11:05:50 CEST 2012
* [.] PVS: Git project reorganisation
  * [ ] Move all projects
  * [X] Update .git/config on local copies
    * [X] vim_config
    * [X] ssh_config
  * [ ] create new projects
    * [ ] General config (scripts and files)
    * [ ] Key discovery
    * [ ] GRE
    * [ ] Inhibitor pattern mining
* [.] JAK2
  * [X] create labels
  * [ ] add lipophilic information
  * [ ] add corresponding label

Fri Jun  8 09:06:26 CEST 2012
* [X] ERC report
  * [X] write information
  * [X] send to Samrat
* [X] JAK2
  * [X] Filtering atoms
    * [X] implements different filters
    * [X] write the datasets according to it
  * [X] Convert the new datasets and the jak2 dataset
  * [X] In jak2 code, correctly read the data
* [.] Various
  * [O] Download interesting courses
    * [X] Automata
    * [X] Computer vision
    * [X] Introduction to logic
  * [ ] Notes on the key discovery problem
    * [ ] look at the 2006 paper, Gordian approach
    * [ ] look at the VLDB 2010 paper
    * [ ] notes on the related work
  * [ ] Git projects on PVS
    * [ ] a project for KeyDiscovery
    * [ ] a project for Pattern-based regression

Thu Jun  7 10:11:40 CEST 2012
* [X] JAK2
  * [X] OpenBabel integration
    * [X] For new datasets
      * [X] read input data files
        * [X] atoms and bonds information
        * [X] molecules information
      * [X] write in jak2 files
        * [X] target_values
        * [X] conformations

Wed Jun  6 10:16:56 CEST 2012
* [X] JAK2
  * [X] Check results from various datasets
  * [X] re-run tests on datasets
    * [X] update scripts
    * [X] run them
  * [X] Start OpenBabel integration
    * [X] compile and install openbabel
    * [X] create a project including OpenBabel
* [o] Various
  * [X] Notes on the key mining problem

Tue Jun  5 10:42:22 CEST 2012
* [X] JAK2
  * [X] update the script parameters.sh
  * [X] update jak2 code to consider new settings
    * [X] use log
    * [X] cross-validation fold number
    * [X] re-add discretization parameters
  * [X] update the script run.sh
  * [X] write a script to run some tests for evaluating the impact of parameters
* [X] Skype Bruno - Bertrand
* [X] Mails
  * [X] to Pascal
  * [X] to Yoann

Mon Jun  4 09:54:03 CEST 2012
* [X] JAK2
  * [X] memory problem
    * [X] manually refining a pattern
    * [X] create new example
    * [X] refine patterns until a certain number of patterns
  * [X] preprocessing data
  * [X] filtering atoms when reading the data
* [o] Other
  * [X] mail to Bertrand and Bruno

dimanche 3 juin 2012, 15:25:48 (UTC+0200)
* [X] JAK2
  * [X] in Cross Validation, create an empty pattern

Wed May 30 09:03:48 CEST 2012
* [X] UM3
  * [X] print:
    * [X] train tickets
    * [X] dossier
    * [X] information on MISASHS
    * [X] slides
    * [X] information on various projects
* [.] JAK2
  * [X] print node information (description)
  * [ ] Scripts to run on JAK2 data (2-step refinement)
  * [ ] Scripts to run on other datasets (2-step refinement)
    * [ ] Preprocessing all files (for regression)
    * [ ] Creating training/test datasets

mardi 29 mai 2012, 08:30:41 (UTC+0200)
* [X] UM3 presentation
  * [X] check the content
  * [X] send it
* [X] Skype Meeting
  * [X] Results
    * [X] on other datasets
      * [X] Results to analyse and compare
      * [X] Do it with the other datasets
      * [X] Comparison
        * [X] what is the measure used in the paper?
    * [X] 2-step refiner
      * [X] no statistics on the failing
      * [X] new strategy?
        * [X] not storing the patterns
        * [X] looping over the patterns before the ligands

lundi 28 mai 2012, 09:16:03 (UTC+0200)
 * [X] Audition UM3
   * [X] Discours slide par slide
 * [X] JAK2
   * [X] when does a pattern is destructed?
   * [X] pattern selection in the query engine

Sat May 26 10:28:03 CEST 2012
* [X] jak2
  * [X] preprocess 5 datasets
  * [X] run tests on explorer
    * [X] 1-step refiner
    * [X] 2-step refiner
* [X] Work on memory consumption issues
  * [X] Remarks:
   - replaced the testCollection by the DBWrapper
   - patterns and corresponding tests are generated, but they are not selected ? --> resulting DT only contain the root
   - when counting the nb of patterns generated and destructed, there is a gap : not all patterns are destructed, where they should be destructed just before creating a new one
* [X] change audition presentation

Fri May 25 14:26:33 CEST 2012
* [ ] take into account Pascal and Sandra's remarks on presentation
* [o] work on jak2 problem
  * [X] correct it
  * [X] push the code
  * [X] run tests on Ronan's data (2-step refiner) on pinacs
* [ ] mail Yoann
* [X] print train tickets
* [X] print 2 papers from Bertrand

Thu May 24 07:49:05 CEST 2012
* [X] Paper
  * [X] read paper
    - QSAR based on the similarity between 3D structures (molecule kernel, independent of the spatial prealignement of the compounds)
    - Kernel + potential SVM
    - Relying on pairs of ordered bipods (V-shaped subfragments)
    - Related work compared with the approach:
      - Hologram QSAR (HQSAR) --> divides each molecule into a set of overlapping structural fragments and uses their frequency as 2D descriptors
      - Comparative molecular field analysis (CoMFa) --> 3D QSAR, based on the idea similar steric and electrostatic fields of molecules lead to similar activity. Requires the spatial-imposition of the molecular structures, which needs expert knowledge and is time-consuming and might introduce user-bias
      - CoMSIA (comparative molecular similarity index analysis) --> altervative method to CoMFa
      - GRIND method: 3D QSAR --> only applicable for a set of compounds sharing a common scaffold
      - EVA (eigenvalue analysis)
    - molecular alignment = rigid transformation of the coordinates of a molecule
    - DATASETS
      - Fontaine (classification): 435 molecules (156 low-, 279 high-activity)
      - ACE (regression): 114 molecules
      - AChE (regression): 111 molecules
      - BZR (regression): 163 molecules
      - DHFR (regression) 397 examples
  * [X] are datasets publicly available?
  * [X] how the paper is cited?
    * [X] 9 times
* [X] Experiments
  * [X] Compare with results in paper
* [X] UM3
  * [X] send pdf document
  * [X] send email to Christian Lavergne
  * [X] send email to Nicolas (infos)
  * [X] répondre Bruno

Wed May 23 08:00:15 CEST 2012
* [O] UM3
  * [X] sélectionner publications
    * [X] lettre signée avec résumé
    * [X] mail Nicolas (nombre d'exemplaires, thèse?)
  * [.] finir présentation
    * [X] trouver et stocker exemples de présentations
    * [ ] enseignement
    * [ ] recherche
  * [X] Mail Sandra : utile de contacter Christian Lavergne?
* [X] summary of last meeting to Thomas
  * [X] mail
* [o] JAK2
  * [O] compare with the results of the genetic algo paper
    * [X] in reading the input data, change IC50 to 1 or 0 whether it is above/below the given threshold --> 0 is inactive, 1 is active
    * [X] remove log(IC50)
    * [ ] note results in classification
    * [X] new Test with memory problem fixed
      * [X] on Ronan data --> pinacs
      * [X] on bzr_3d data: with 1-step refiner --> test in a classification context
  * [ ] look at http://pubs.acs.org/doi/abs/10.1021/ci800144y, do they have a publicly available dataset? 
  * [X] memory problem
    * [X] replace shared_ptr<pattern> by weak_ptr<pattern> --> not possible without more work
    * [X] found and fixed

Tue May 22 09:22:11 CEST 2012
* [X] JAK2
  * [X] test with/without keeping alternative patterns
    * [X] Result: no difference at all between keeping and not keeping intermediate patterns
  * [X] explicitely clear the test collection of a node when it has been finished (in mips)
    * [X] clear the container (DBWrapper == TestCollection) when all tests have been checked
    * [X] print memory usage before and after --> no effect at all on the destruction of the patterns
    * [X] find a way to clear the patterns when the test collection is cleared
      * [X] Problem: in a test, the pattern is associated with a boost::shared_ptr --> maybe even when the test collection is cleared, the patterns are pointed by more than one object (i.e., by at least another object than the test itself) --> yes, there are several references to the same pattern (2 at the beginning, 3 after).
      * [X] Is there a case where there is only 1 reference to a pattern? --> apparently not. So, why patterns are sometimes destructed?
      * [X] in removeUsedTestGenerator.hpp it seems that the test collection of a parent node is copied and passed to children? --> not used for jak2
      * [X] print the number of references to a pattern
      * [X] in embeddingTest (jak2), create a delete function
      * [X] is the number of references to a pattern the same for 1 and 2-step refiners?

Mon May 21 09:05:03 CEST 2012
* [X] UM3
  * [X] mail Nicolas 
  * [X] préparer dossier pdf
* [X] Jak2
  * [X] Code
    * [X] Test the memory consumption with 1-step refiner
    * [X] Create a destructor for the patterns and print something when used
    * [X] Speak with Thomas
  * [X] Answer Bruno's email

Mon May 14 10:41:56 CEST 2012
* [X] Mail Pascal et Sandra
* [X] Bertrand and Ronan
  * [X] Read Muggleton 2008 paper (in particular, the data description)
  * [X] Discuss it with Thomas
* [o] UM3
  * [X] Mail to Nicolas
  * [ ] Create PDF document
* [ ] Mail Jan
  * [ ] results of experiments (how long for learning the first DT?)
  * [ ] Audition in Montpellier
* [ ] Mail Yoann
* [ ] JAK2 project
  * [ ] convert the datasets of JCIM2003 (Spline-fitting with a genetic algorithm, Sutherland et al...)
  * [ ] look at what has been done by Muggleton
  * [ ] integrate angles in constraints?
* [X] Text prediction: 
  * [X] execute FPGrowth

Wed May  9 08:52:18 CEST 2012
* [X] select best patterns in trees
* [X] push new code on git
* [X] put experiments on svn
* [ ] look at other datasets with 3D information
* [X] look at the impact of the equalWidth discretizer
* [X] add negative examples to Muggleton's data (by removing atoms of the pharmacophore) --> cancelled, we will wait for another dataset
* [ ] change format of experiments summary --> in csv
* [X] run with no grounded at labels and 2-levels deep lookahead

Tue May  8 08:56:23 CEST 2012
* [X] give all the embeddings of one given pattern in one given ligand
  * [X] in subgraph isomorphism, store all the possible embeddings
  * [X] in ConjunctiveSubgraphIsomorphism, count and write (in an outstream) all the possible embeddings
* [X] run DT learner on Muggleton's data
  * [X] equivalent of IC50 value?

Mon May  7 10:27:15 CEST 2012
* [X] Convert Muggleton data (eclipse)
  * [X] read atoms and bonds
    * [X] add charge information --> everything to 0
    * [X] add aromaticity information
  * [X] write cvs files
* [X] Download courses
* [X] Experiments: run with equalWidth interval distances, and smaller interval size, and overlapping intervals (1, 2, 3, 4)
* [X] Mail Bruno and Bertrand
* [X] install mail kuleuven on phone
* [ ] Interesting paths in decision trees
* [X] mutt configuration
  * [X] added gmail address
* [X] separate jak2 code for experiments and coding
* [X] move scripts and results in the new folder for experiments

Fri May  4 10:46:51 CEST 2012
* [X] phone
  * [X] charge
  * [X] files
* [X] Answer mail Sandra, Pascal 
* [X] Mail UM3
* [X] Printing 2 chapters from Graph Theory book
* [X] Start transforming Muggleton's dataset

jeudi 3 mai 2012, 14:31:16 (UTC+0200)
* [X] fix problem with discretizer type preprocessor
* [X] added parameters for overlapping interval number in settings
* [X] fix problem with ROC computing

mercredi 2 mai 2012, 09:42:11 (UTC+0200)
* [X] Mail réponse Sandra
* [X] Put first results on svn repository
* [X] Discretization based on equal-width intervals
  * [X] allow overlapping intervals for all distance discretizers
    * [X] method discretize returns a list of intervals
    * [X] implement discretize for
      * [X] Geometric discretizer (only for 1 overlapping interval)
      * [X] EqualWidth discretizer
    * [X] Check if the new discretization works well with preprocessings (distance lists, etc.)
* [X] Look for other measures for test selection in the DT learner

mercredi 25 avril 2012, 15:01:54 (UTC+0200)
* [O] Implement and count ROC-like value
  * [X] in 1 DT, count all couples of ligands
  * [X] in 1 DT, count only couples of consecutive ligands
  * [ ] extend to the cross-validation (average)
  * [X] extend to the cross-validation (taking into account all possible ligands)
  * [X] integrate into the scripts
    * [X] java post-processer read and write
* [X] Logs all experiments on trantor -> add the date
* [X] Print the settings in the DT log
* [X] Print the predictions in the DT log
* [X] In HTML summary of experiments
  * [X] add the date
  * [X] add links to the log files
* [X] Add to experiment scripts:
  * [X] an argument to choose the type of distance discretization
  * [X] an usage section
* [ ] Convert Muggleton data into our format + try DT learner
* [X] Distance discretization based on equal-width intervals
  * [X] allow an overlapping of intervals
* [ ] Count embeddings of a pattern in a ligand (in each conformation) --> to be used in Leander visualization tool
* [ ] Documentation Doxygen

* [ ] AKDM3 
  * [ ] mail to organizers
  * [ ] mail to Pascal, Sandra
* [o] JAK2  
  * [ ] run tests
  * [X] add the variance reduction and information gain (avg and standard deviation) in:
    * [X] log
    * [X] scripts
    * [X] java post-processing
  * [X] log global results in a HTML page
* [X] install jak2 code on explorer

jeudi 19 avril 2012, 09:35:36 (UTC+0200)
* [X] Mail Hassan, Yoann

mercredi 18 avril 2012, 10:06:54 (UTC+0200)
* [X] List of things to read
* [ ] Message for Ronan?
* [ ] Data on Caen computer
* [o] Mails:
  * [X] Pascal, Sandra 
  * [ ] AKDM3

vendredi 13 avril 2012, 09:21:17 (UTC+0200)
* [O] Code
  * [X] external class cross-validation
    * [X] log results
		* [X] log the folds
	* [X] in settings, add the interesting leaves IC50 threshold
  * [ ] have results for interesting leaves only (IC50 < 100)
  * [X] external class find embeddings
  * [X] external class leave sorter
  * [X] Experimental protocol
    * [X] script to:
      * [X] change the settings in settings.hpp 
			* [X] run the code (compilation+running)

mardi 10 avril 2012, 08:10:51 (UTC+0200)
* [ ] contract --> send to Annemie Dumoulin
* [X] email with address and bank account
* [ ] RMSE global in the DT and for interesting leaves only (IC50 < 100)
  * [ ] multiply by the number of examples
* [ ] Benchmarks
  * [ ] find datasets
  * [ ] ask Leander for datasets and methods to compare with
* [X] mail Nicolas

lundi 9 avril 2012, 10:48:46 (UTC+0200)
* [X] get clang_complete plugin working
* [X] dr
* [X] pattern matcher code
  * [X] use connected_components from boost
  * [X] optimize the methods using the nth vertex
    * [X] use the access to vertices and edges g[edgeOrVertex]
      * [X] in vertexList option

Fri Apr  6 11:00:02 CEST 2012
* [ ] Pattern matching with unconnected graphs (conjunctions of graphs)
* [X] Log embeddings
  * [X] as a post-processing
  * [X] INPUT: pattern / OUTPUT : list of embeddings
  * [X] in OUTPUT log, print a pattern with the code of the node it is associated with (optional)
* [X] Pattern OUTPUT
  * [X] as a list of logical conjunctions (e.g., GroundedAt(X,4,2.3,1),InDistance(X,Y,6.7),...)
* [o] PATTERN INPUT/OUPUT
  * [X] GraphViz output
* [.] TESTS
  * [o] Allow a 10-fold cross-validation
    * [X] automatic random segmentation of the dataset
    * [ ] running and storing the correct information
  * [ ] As INPUT : the pattern language wanted for the execution
  * [ ] As OUTPUT :
    * [ ] average time (and +/- standard deviation?)
    * [ ] RMSE (rooted mean squared error) --> in multi-instances settings, use the conformation with the smallest IC50 value
    * [ ] average tree size +/- standard deviation
    * [ ] log everything
      * [ ] the datasets resulting from the cross-validation?

Wed Apr  4 09:41:01 CEST 2012
* [ ] Document about list of experiments
  * [ ] RMSE instead of MSE
  * [ ] average time on cross-validation and size of the tree

Tue Apr  3 09:18:28 CEST 2012
* [X] Meeting with Jan, Thomas, Bertrand
* [X] Updating the code	on local machine
* [X] installing QT-creator on local machine
* [X] Administration
  * [X] Groot Begijnhof
    * [X] contract Groot Begijnhof
    * [X] inventory
    * [X] contacting bank for the bank transfer

lundi 26 mars 2012, 08:49:12 (UTC+0200)
* [ ] Dossier candidature
  * [ ] ajouter école de printemps?
  * [ ] 
* [X] Mail Thomas accommodation
* [ ] Code jak2
* [X] Mail Mathieu Roche

dimanche 25 mars 2012, 14:03:27 (UTC+0200)
* [o] Dossier UM3
  * [ ] Dossier enseignement
    * [ ] Détail Fouille de Texte : ajouter lien vers supports de cours
    * [ ] Ajouter Béziers et insister sur 'je connais ...'
    * [ ] Parler des différentes formations
  * [X] Dossier recherche
    * [X] Finir texte
  * [o] Divers
    * [X] Ajouter jury de thèse en page 2
    * [X] Ajouter Leuven en page 1 (CV)
    * [ ] Ajouter intervenants

vendredi 23 mars 2012, 08:09:23 (UTC+0100)
* [o] Dossier candidature MCF UM3
  * [X] Relire projet enseignement
  * [X] Envoyer à Sandra
  * [ ] Ecrire projet de recherche
  * [ ] Envoyer à Pascal
* [X] Corrections Fouille de texte
* [ ] Papier AKDM
* [X] Rappeler Bruno et Bertrand (lettre de recommandations)

mercredi 21 mars 2012, 08:35:04 (UTC+0100)
* [X] Final version of AKDM paper
* [X] mail Maguelonne
* [X] mail Mathieu
* [X] mail Marc
* [X] mail Enseignements UCBN
  * [X] envoyer CV
* [X] List of experiments on Jak2
  * [X] Ronan
    * [X] One and Two-levels deep refinement strategy
    * [X] Distance upper limit --> TO IMPLEMENT
    * [X] Finer discretization on distances
    * [X] With/without a strategy for reducing IC50 value in Yes branches --> TO IMPLEMENT
  * [X] Paper
    * [X] IC50 value, log(IC50), rank
    * [X] Prediction accuracy (with training and test datasets)
    * [X] Different discretization methods:
      * [X] Distances
        * [X] based on a geometric series interval bounds (current)
        * [X] based on equal-frequency intervals
        * [X] based on equal-length intervals
      * [X] Coordinates
        * [X] different values on boxes size
        * [X] space-segmentation methods
    * [X] Pre/Post-pruning parameters in the decision tree?
* [X] Jak2 code
  * [X] distance upper limit
  * [X] decreasing IC50 value
* [X] Summary document on Jak2 project for Bruno
* [X] Dossier MCF : Enseignement
* [X] Corrections AKDM Sandra

dimanche 18 mars 2012, 09:55:10 (UTC+0100)
* [X] Mail KUL (admin)
* [X] Dossier UM3
  * [X] CV
  * [X] Préparer mail
  * [X] Envoyer mail
* [X] Code
  * [X] supprimer vertexList

lundi 12 mars 2012, 08:45:50 (UTC+0100)
* [X] Mails:
  * [X] Accommodation Leuven
  * [X] Bertrand : plan présentation
  * [X] Thomas : 1 level-deep
  * [X] Sandra : recommandation letter
  * [X] Admin : GREYC web site

dimanche 11 mars 2012, 15:14:17 (UTC+0100)
* [X] mail Sandra
  * [X] corrections lettre + remarque
  * [X] n° de poste
* [X] Cours 3 Fouille de texte

mercredi 7 mars 2012, 17:21:53 (UTC+0100)
* [X] Presentation preparation for IATE on 2012-03-09
  * [X] S6: better link between association rules and FCA?
  * [X] S6: "the attributes are directly viewed as itemsets" --> "the attributes are directly viewed as items"
  * [X] S7: not "translating the FCA problem into a frequent itemset mining problem"?
  * [X] S7: automatic ontology elicitation?
  * [X] S30: put an image from the last presentation?
  * [X] S32: first line : ????! :)
  * [X] At the beginning, make a slide on the itemset mining problem (quite simple) in order to better see the translation into FCA
    * [X] Quick presentation of the problem --> example(s)
    * [X] Show the link with knowledge acquisition
  * [X] Implications in a concept lattice
    * [X] have a good understanding of it: e.g., what happens with dog?
    * [X] introduce the fact that association rules can also provide this type of rules, but with a less "crisp" notion of implication
  * [X] How to introduce sequential pattern mining?
  * [X] How to introduce contextual pattern mining?
    * [X] think about the exploitation of contextual pattern mining in itemsets : what can it give that classical pattern mining can't? Domain knowledge can be exploited in order to build the context hiearchy
  * [X] When showing items in sequences that are related to originally numerical attributes, speak about fuzzy logic and pattern mining techniques?
  * [X] Show one context hierarchy in the slides?
  * [X] Globally highlight the fact that extracting patterns is usually very good when dealing with experts : they are interpretable (like the original idea of fuzzy logic?.. :) )
  * [X] Find a concrete example of the fact that contextual information can be seen at a different level than attributes-values
  * [X] when dealing with sequential patterns, mention that it is about dealing with structured data (sequences, graphs, etc.), which is a more complex domain than itemsets, but that can model numerous real-world domains (e.g., time-related domains, bioinformatics, texts, chemistry, etc.). Mention the fact that I already have worked on numbers of them.
  * [X] Two directions: from pattern mining to knowledge and "from knowledge (ontologies, taxonomies) to patterns to applications"
    * [X] Direction 1
      * [X] fastly provide an ontology based on concepts taht can be observed within the data (then validation)
    * [X] Direction 2
      * [X] being able to extract patterns at a given level of a taxonomy or directly exploiting the different levels of a taxonomy (without a specified one)
      * [X] use the obtained knowledge to build the contexts that will be used in order to obtain precise results

mardi 6 mars 2012, 15:16:38 (UTC+0100)
* [X] Presentation in chemoinformatic group
* [X] Document for GREYC administration (mission document)
* [ ] Thomas
  * [ ] code running?
  * [ ] meeting with Ronan? Coming to Caen?
* [o] Leuven
  * [ ] registration on web site
  * [X] username and password
  * [X] documents
    * [X] CV
    * [X] passport
    * [X] diploma
* [X] Mail Hugo
* [ ] Teaching
  * [ ] Lectures
    * [ ] 
  * [ ] Practical work
* [X] GREYC web site

lundi 5 mars 2012, 15:52:40 (UTC+0100)

  * [X] Arbre de décision
    * [X] Exemple + explications
    * [X] Avantage : problème de seuil
  * [X] Exemple motif --> animation
    * [X] créer images pour BD
    * [X] créer images pour motif
  * [ ] Mettre en forme
  * [X] Remplacer la section "résultats expérimentaux" par autre chose ?
  * [X] To add
    * [X] logos on slide 1
    * [X] slide numbers
    * [X] unité pour l'IC50 (nM)

vendredi 2 mars 2012, 09:34:59 (UTC+0100)
* [X] mail réponse Bertrand
* [X] mail Thomas
* [X] any idea for pattern matcher time?
* [X] advances on presentation

jeudi 1 mars 2012, 15:36:06 (UTC+0100)

* [X] réunion Ronan CERMN
* [X] mail Bertrand
* [X] Code
  * [X] problem with distances
  * [X] check and insert rules for hybridization
  * [X] logarithmic IC50 value
* [ ] Having jak2 and mips code running on laptop

mercredi 29 février 2012, 14:52:19 (UTC+0100)

* [X] Code
* [X] Questions Ronan
* [X] Mail Thomas
* [X] Mail Bruno (déplacement Paris)


vendredi 24 février 2012, 08:43:10 (UTC+0100)
* [X] cours
	- [X] imprimer documents utiles
	- [X] mail Gaël Dias, Frédérick
	- [X] plan 1er cours + demander énoncer TP Nicolas


= GDR I3 presentation: outline =
* I. Contexte chémogénomique
	- Affinité protéique
	- Voies de dégradation
	- Application : inhiber JAK, découvrir des ligands
	- Montrer une image qui montre l'affinité d'une protéine et d'un ligand (conformation)
* II. Information recherchée (Motifs recherchés)
	- Type de motifs
	- Arbre de décision avec motifs
	- Aspect multi-instances (différentes conformations --> 1 slide : un ligand satisfait un motif si une des conformations satisfait le motif)
* III. Méthode d'extraction
	- Ne pas focaliser dessus pour le 6 mars
* IV. Résultats expérimentaux
	- Temps en fonction de certains paramètres
	- Avec ou sans utilisation de la stratégie de refinement sur le nouveau sommet généré
	- Nombre de motifs générés
* V. Conclusion et perspectives
	- Outil de visualisation des motifs dans les conformations pour les experts, outil de paramétrage
	
mardi 28 février 2012, 10:36:49 (UTC+0100)
* [X] Finir Cours + TP
  * [X] cours
    * [X] logos
    * [X] corrections
    * [X] other exercices?
    * [X] avantages des motifs : interprétabilité, tiennent pleinement compte de l'aspect séquentiel des données textuelles, peuvent être considérés comme de nouveaux descripteurs d'un texte (classification, ...)
    * [X] notation
  * [X] TP
    * [X] pas à pas :
      * [X] compiler le code
      * [X] extraire sans filtrage et lemmatisation
      * [X] extraire sans filtrage et avec les étiquettes grammaticales
      * [X] traduire les motifs
      * [X] extraire avec filtrage grammatical
      * [X] extraire avec lemmatisation
* [O] JAK2 Data
  * [X] create list of distances (Java)
  * [X] create list of coordinates (on x, y , z) (Java)
  * [X] create histograms (Weka, R, Gnuplot, ?)
  * [.] write short document
    * [ ] different parameters for the discretization
    * [X] histograms and data
      * [X] minimum and maximum values, mean value, etc.
      * [X] density histograms
    * [ ] how to select the right parameters?
* [X] email to Ronan --> ok for thursday
  * [X] first patterns (decision tree)
  * [X] book + usb key
  * [X] data to tag
  * [X] H-bond donor / acceptor?
